---
title: CrossTalkeR Cookbook
author:
- name: James S. Nagai
  affiliation: Institute for Computational Genomics,
               Faculty of Medicine,
               RWTH Aachen University,
               Aachen, 52074 Germany
- name: Vanessa Kl√∂ker
  affiliation: Institute for Computational Genomics,
               Faculty of Medicine,
               RWTH Aachen University,
               Aachen, 52074 Germany
- name: Ivan G. Costa
  affiliation:
     - Institute for Computational Genomics,
       Faculty of Medicine,
       RWTH Aachen University,
       Aachen, 52074 Germany
output:
     html_document
vignette: |
      %\VignetteIndexEntry{CrossTalkeR-cookbook}
      %\VignetteEngine{knitr::rmarkdown}
      %\VignetteEncoding{UTF-8}

---


# Version Info
```{r vignete0, eval = FALSE}
suppressPackageStartupMessages({
  require(CrossTalkeR)
})
suppressPackageStartupMessages({
  require(igraph)
})
suppressPackageStartupMessages({
  require(ggraph)
})
suppressPackageStartupMessages({
  require(ggplot2)
})
suppressPackageStartupMessages({
  require(EnhancedVolcano)
})
```

**R version**: `r R.version.string`

**Package version**: `r packageVersion("CrossTalkeR")`

# Generate Report Example
In our vignette we provide examples on how to analyse cell interactions from a human myelofibrosis single cell RNA-seq dataset.


```{r vignete1, results="hide", warning=FALSE,message=F, output=FALSE, eval = FALSE}
data(CTR)
data(EXP)
paths <- list('CTR'=CTR,
              'EXP'=EXP)

genes <- c('TGFB1|L')
output <- system.file("extdata", package = "CrossTalkeR")
data <- generate_report(paths,
                        genes,
                        out_path=file.path(output),
                        threshold=0,
                        out_file = 'vignettes_example.html',
                        output_fmt = "html_document",
                        report = TRUE)
```

# Individual Visualization

## CCI
```{r vignete2, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8, eval = FALSE}
plot_cci(
  graph = data@graphs$CTR,
  colors = data@colors,
  plt_name = "Example 1",
  coords = data@coords[V(data@graphs$CTR)$name, ],
  emax = NULL,
  leg = FALSE,
  low = 0,
  high = 0,
  ignore_alpha = FALSE,
  log = FALSE,
  efactor = 8,
  vfactor = 12
)
```

## Sankey plot
```{r vignete4, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=15,fig.height=10, eval = FALSE}
plot_sankey(
  lrobj_tbl = data@tables$EXP_x_CTR,
  target = c("TGFB1|L"),
  ligand_cluster = NULL,
  receptor_cluster = NULL,
  plt_name = "TGFB1"
)
```

## Report Samples

[Single](Single_vignettes_example.html)

[Comparative](Comparative_vignettes_example.html)


# Session information {.unnumbered}

```{r sessionInfo, echo=FALSE, eval = FALSE}
sessionInfo()
```
