---
title: CrossTalkeR Example - Human Myelofibrosis
author:
- name: James S. Nagai
  affiliation: Institute for Computational Genomics,
               Faculty of Medicine,
               RWTH Aachen University,
               Aachen, 52074 Germany
- name: Nils B. Leimkühler
  affiliation: Department of Hematology and Stem Cell Transplantation,
               University Hospital Essen,
               Germany
- name: Vanessa Klöker
  affiliation: Institute for Computational Genomics,
               Faculty of Medicine,
               RWTH Aachen University,
               Aachen, 52074 Germany
- name: Michael T. Schaub
  affiliation:
     - Department of Computer Science,
       RWTH Aachen University,
       Aachen, Germany
- name: Rebekka K. Schneider
  affiliation:
    - Department of Hematology,
      Erasmus Medical Center,
      Rotterdam, 3015GD,
      the Netherlands
    - Department of Cell Biology,
      Institute for Biomedical Engineering,
      Faculty of Medicine,
      RWTH Aachen University,
      Aachen, 52074 Germany
    - Oncode Institute,
      Erasmus Medical Center,
      Rotterdam, 3015GD,
      the Netherlands
- name: Ivan G. Costa
  affiliation:
     - Institute for Computational Genomics,
       Faculty of Medicine,
       RWTH Aachen University,
       Aachen, 52074 Germany
output:
     html_document
vignette: |
      %\VignetteIndexEntry{CrossTalkeR-HumanMyfib}
      %\VignetteEngine{knitr::rmarkdown}
      %\VignetteEncoding{UTF-8}
---
# Cell-Cell Communication Analysis of Human Myelofibrosis scRNA-seq Data Using CrossTalkeR

Here, we demonstrate the usage of CrossTalkeR on the case study Myelofibrosis scRNA-seq data set from the paper. The predicted ligand-receptor interactions used in this tutorial are provided within this package. These differ sightly from the results presented in the paper, since they were recomputed using the cellphoneDB Method with the liana package and the consensus interaction database. If you want to learn more about how to run liana to use it with CrossTalkeR, check out this turorial: LINK.

The scRNA-seq data contains two conditions: control (CTR_LR.csv) and disease condition (EXP_LR.csv). We first show how to run CrossTalkeR and then go over some of the results, that are also provided inside the resulting CrossTalkeR report

## Running CrossTalkeR

First, we need to import the CrossTalkeR package and define a named list of either the paths to the Ligand-Receptor interaction tables, or directly dataframe objects for each condition of interest. Please note that it is possible to run CrossTalkeR with more than two conditions. (TODO: hide imports needed only for vis later stringr igraph) Here we load the files from within the package. We can further define a path where we like to save our results (output) and run CrossTalkeR with the generate_report() function:
```{r, warning=FALSE,message=F, output=FALSE, include = FALSE}
library(CrossTalkeR)
library(igraph)
library(stringr)
library(ggplot2)
library(colorBlindness)
library(dplyr)
```

```{r, warning=FALSE,message=F, output=FALSE, eval = FALSE}
library(CrossTalkeR)
library(igraph)
library(stringr)

paths <- c('CTR' = system.file("extdata",
                               "CTR_LR.csv",
                               package = "CrossTalkeR"),
           'EXP' = system.file("extdata",
                               "EXP_LR.csv",
                               package = "CrossTalkeR"))

output <- system.file("extdata", package = "CrossTalkeR")
data <- generate_report(paths,
                out_path=paste0(output,'/'),
                out_file = 'vignettes_example.html',
                output_fmt = "html_document",
                report = TRUE)

```

Besides passing the paths to the interaction tables and the output folder, we here also define:
    out_file -> A suffix for the result reports
    output_fmt -> The file type of the result reports
    report -> A boolean value if the reports should be created at all

As result of the successful execution, we receive three seurat objects (LR_data.Rds, LR_data_step2.Rds and LR_data_final.Rds), as well as two html-reports(SingleCondition.html and ComparativeCondition.html). The LR_data_final.Rds contains all results produced by CrossTalkeR and the two html-reports contain the visualization of the results inside the R object. It is possible to load the results of the CrossTalkeR analysis from the "LR_data_final.Rds" R-object and produce the plots separately. In the following we are going to take a look at different plots within the two reports, step by step.

## Single Condition report

The first report contains the results for each analysed condition considered separately. In our case in each section is a plot for the control condition (CTR) and the disease condition (EXP). There are two main sections in the report. The first one only deals with cell-cell-interactions and the second with cell-gene-interactions, which here means an analysis at the ligand and receptor gene level.

### Cell-Cell-Interaction analysis results

The cell-cell-interaction (CCI) graph plots focus on the interactions between the different cell types within the data. We focus on three measures inside this plot:
    1. Percentage of interactions indicated by the opacity of the arrows
    2. Edge weight (sum of single L-R interaction scores between celltype pairs) indicated by the edge color
    3. Pagerank (Node importance) indicated by the node size

After loading the result data from the "LR_data_final.Rds", we can plot the cell-cell interactions with the plot_cci() function. Here we show a plot for each analysed consition:
TODO: show CTR and EXP next to each other
```{r vignete2, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8, fig.show="hold" }
data <- readRDS(system.file("extdata", "LR_data_final.Rds", package = "CrossTalkeR"))
p1 <- plot_cci(graph = data@graphs$CTR,
        colors = data@colors,
        plt_name = 'Control',
        coords = data@coords[V(data@graphs$CTR)$name,],
        emax = NULL,
        leg = FALSE,
        low = 0,
        high = 0,
        ignore_alpha = FALSE,
        log = FALSE,
        efactor = 8,
        vfactor = 12,
        pg = data@rankings[["CTR"]]$Pagerank)

p2 <- plot_cci(graph = data@graphs$EXP,
        colors = data@colors,
        plt_name = 'Disease',
        coords = data@coords[V(data@graphs$EXP)$name,],
        emax = NULL,
        leg = FALSE,
        low = 0,
        high = 0,
        ignore_alpha = FALSE,
        log = FALSE,
        efactor = 8,
        vfactor = 12,
        pg = data@rankings[["EXP"]]$Pagerank)

```
In the control plot we can see that a high proportion of interactions are MSCs talking with themselves as well as MSC and Fibroblasts talking with each other. These interaction have also a higher weight, as can be seen by the edge color. The most important node by pagerank is the MSC node. When analysing the plot of the disease condition, we see again a high proportions of interactions are self-talking MSCs and the most important node by pagerank is again the MSC node. Here the MSC are talking more to Megakaryocytes. To really compare the two conditions we recommend using the Comparative report results.

### Principal Component Analysis plots on gene-gene interaction level
To find relevant ligand and receptor genes inside the network, we performed a principal component analysis (PCA) on the network topological measures (Listener, Influencer, Mediator, Pagerank). The results of the PCA are included in form of a plot, as well as inside a table. In the plot top ligand and receptors are indicated with labels and arrows indicate the directions of importance for each topological measure in the 2D space. Here we show the plots of the first 2 dimensions for the CTR and EXP condition in our data.

TODO change plot function and add description
```{r vignete3, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
print(plot_pca_LR_comparative(data, "CTR_ggi", dims = c(1, 2), ret = T, ggi = TRUE, include_tf = FALSE))
print(plot_pca_LR_comparative(data, "EXP_ggi", dims = c(1, 2), ret = T, ggi = TRUE, include_tf = FALSE))
```
Based on the PCA plots we can get a first impression on which gene might be important in the interactions and also in which role. As example in the control condition the Ligands APP and MDK in MSCs, and COL1A1 and TGFB1 in Fibroblast seem to be important influencers in the network. On the other hand the receptors LRP1 and ITGB1 in MSCs seem to have a high overall importance (pagerank) and also seem to be important listeners. The same way it is possible to interpret the plot of the disease condition.

Since only a few genes are highlighted within the PCA plot, we also include the table with the results of the PCA in our reports. The tables are interactive and it is possible to sort by the different columns. As example if we want to find the top receptors that have the role of a listener, we can simply sort the "Listener" column by decreasing value and check the entries at the top. Mark that if we are only analysing ligand-receptor interactions we don't receive any mediator genes, since these are pairwise interactions.

Tables Ranking of genes based on the analysed topological measures NOT RENDERED YET
```{r vignete4, warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
all_data <- data
out <- c()
out1 <- list()
for (i in 1:length(names(all_data@pca))) {
  curr <- names(all_data@pca)[i]
  if (!str_detect(curr, '_x_', negate = FALSE) & str_detect(curr, '_ggi', negate = FALSE)) {
    rmd_title <- paste0(curr, '_tbl')
    rmd_title1 <- paste0(curr, '_pca')
    title <- paste0('\n\n#### Table PCA CGI ', curr)
    knit_expanded <- paste0("```{r , results='", rmd_title, "', echo=FALSE, warning=FALSE}\n\n
         pca <- all_data@pca[['", curr, "']]$x[,1:2]
         pca[,1] <- round(pca[,1],6)
         pca[,2] <- round(pca[,2],6)
         ranking <- cbind(all_data@rankings[['", curr, "']][,2:dim(all_data@rankings[['", curr, "']])[2]],pca[all_data@rankings[['", curr, "']]$nodes,])
         datatable(ranking,
                 caption='", curr, "',
                 extensions = 'Buttons',
                 options = list(dom = 'Bfrtip',
                                buttons = c('copy',
                                            'csv',
                                            'excel',
                                            'pdf', 'print'),
                               autoWidth = FALSE
                             ))\n\n```")
    out <- c(out, knit_expanded)
    out1[[curr]] <- knit_expanded
  }
}
```

### Most significant deregulated pathway per topological measure
Within the package we also provide a functional analysis on gene sets in form of KEGG pathway enrichment. The genes are grouped based on their topological role, so by listener, influencer or mediator, and also by high pagerank values. We summarize the results as heatmap, with only the significant pathways included (p-val < 0.05).
```{r vignete5, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
all_data <- data
paths <- read.csv(system.file("extdata", "selected_KEGG.csv", package = "CrossTalkeR"), header = T)
for (i in names(all_data@annot)) {
  if (str_detect(i, 'ggi', negate = FALSE) & !str_detect(i, '_x_', negate = FALSE)) {
     plot_deregulated_pathways(data_object = all_data,
                               name = i,
                               title = i)
  }
}

```


## Compared Condition results
CrossTalkeR not only analyzes the conditions individually but also calculates the differences in the cell-cell interactions between the conditions. The results of these analyses are presented in the comparative report.  Here you can find many of the already presented graphs but with additional possibilities of result interpretation. Therefore, we will revisit here several of the plots already described for the individual conditions. For our example we are comparing the disease (EXP) against our control (CTR) condition.

### CCI plot
not only positive weights on edges of graph, but also negative in difference, same for pagerank values on the nodes. recommend to/ possible to remove this information from plot an analyse separately with bar plot on pagerank.
```{r vignete6, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
plot_cci(graph = data@graphs$EXP_x_CTR,
        colors = data@colors,
        plt_name = 'Disease vs Control',
        coords = data@coords[V(data@graphs$EXP_x_CTR)$name,],
        emax = NULL,
        leg = FALSE,
        low = 0,
        high = 0,
        ignore_alpha = FALSE,
        log = FALSE,
        efactor = 8,
        vfactor = 12)

```
Here in our example we see more strong blue arrows, remarkably in MSC-MSC and MSC-Firbroblast interactions. This means these interaction have a higher weight in the control condition, thus a down regulation of these interactions is happening in the disease condition. If we include Pagerank values on the nodes resulting in very small node sizes with large negative values, same explanation as for negative edge weights, higher importance in control condition in this example. We don't see that many orange/red edges in our example but these mean a higher weight up regulation in disease condition.

### PCA plots based on cell types
know PCA plot from single report with ligand receptor genes, also possible to use on cell types on CCI level.
```{r vignete7, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
print(plot_pca_LR_comparative(data,
                        "EXP_x_CTR",
                        dims=c(1,2),
                        ret=T,
                        ggi=FALSE))

```
example not showing all direction arrows, happens if one measure is in the not visible dimension of the reduction, or lost of information due reduction?. still derive information Myeloid cells as mediators in the network and Neural high Pagerank/Listener

### Bar plot pagerank CCI
including the pagerank bar plot, mentioned at CCI graph plot. easier to derive most important nodes for both condition, EXP positive and CTR negative
```{r vignete88, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
for(i in 2:length(names(data@rankings))){
  curr <- names(data@rankings)[i]
  if(str_detect(curr, '_x_', negate = FALSE) & !str_detect(curr, 'ggi', negate = FALSE)){
    signal <- ifelse(data@rankings[[curr]]$Pagerank < 0, 'negative','positive')
    print(ggplot(data@rankings[[curr]], aes(x=Pagerank,y=reorder(nodes,Pagerank),fill=signal))+
        geom_bar(stat="identity")+
        scale_fill_manual(values = c(Blue2DarkOrange18Steps[4],Blue2DarkOrange18Steps[14]))+
        theme_minimal())
  }
}
```
Neural here highest pagerank -> most important node in network for EXP condition, MSC and Fibroblast similar low -> more important in CTR condition

### PCA plots CGI

```{r vignete8, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
print(plot_pca_LR_comparative(data,
                              "EXP_x_CTR_ggi",
                              dims = c(1, 2),
                              ret = T,
                              ggi = TRUE,
                              include_tf = FALSE))
```
Pagerank Log Ratio
```{r vignete9, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
all_data <- data
for (i in 2:length(names(all_data@rankings))) {
  curr <- names(all_data@rankings)[i]
  if (str_detect(curr, '_x_', negate = FALSE) & str_detect(curr, 'ggi', negate = FALSE)) {
    tmp <- top_n(all_data@rankings[[curr]], 20, abs(.data$Pagerank))
    signal <- ifelse(tmp$Pagerank < 0, 'negative', 'positive')
    print(ggplot(tmp, aes(x = Pagerank, y = reorder(nodes, Pagerank), fill = signal)) +
            geom_bar(stat = "identity") +
            scale_fill_manual(values = c(Blue2DarkOrange18Steps[4], Blue2DarkOrange18Steps[14])) +
            theme_minimal())
  }
}
```
Tables Ranking of genes based on the analysed topological measures
```{r vignete10, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
plot_cci(graph = data@graphs$CTR,
        colors = data@colors,
        plt_name = 'Example 1',
        coords = data@coords[V(data@graphs$CTR)$name,],
        emax = NULL,
        leg = FALSE,
        low = 0,
        high = 0,
        ignore_alpha = FALSE,
        log = FALSE,
        efactor = 8,
        vfactor = 12)

```
### Most significant deregulated pathway per topological measure
```{r vignete11, results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
 plot_deregulated_pathways(data_object = data,
                               name = "EXP_x_CTR_ggi",
                               title = "EXP_x_CTR")

```


## Follow-up analysis

It is possible to take a closer look at genes, that were found as important in the communication-networks produced by CrossTalkeR. We can rerun the generatio of the reports with adding a list with genes of interest.
Name both options? with |L and without?
```{r}
genes <- c('TGFB1|L')
```

As result, Sankey plots showing the interactions with the genes of interest are added to the comparative condition report. Here you can see the example Sankey plot for the TGFB1 ligand gene.
```{r , results="hide", warning=FALSE,message=F, output=FALSE,fig.width=8,fig.height=8}
plot_cci(graph = data@graphs$CTR,
        colors = data@colors,
        plt_name = 'Example 1',
        coords = data@coords[V(data@graphs$CTR)$name,],
        emax = NULL,
        leg = FALSE,
        low = 0,
        high = 0,
        ignore_alpha = FALSE,
        log = FALSE,
        efactor = 8,
        vfactor = 12)

```