<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Single Cell Ligand Receptor Downstream analysis â€¢ CrossTalkeR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Single Cell Ligand Receptor Downstream analysis">
<meta name="description" content="This Package allows the network based analysis over ligand receptor network infered data.">
<meta property="og:description" content="This Package allows the network based analysis over ligand receptor network infered data.">
<meta property="og:image" content="/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-black" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">CrossTalkeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/CrossTalkeR.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="articles/CrossTalker_basicusage.html">CrossTalkeR 101</a></li>
    <li><a class="dropdown-item" href="articles/HumanFibrosis.html">Example: Bone Marrow Fibrosis in Human (Leimkuhler et. al., 2021)</a></li>
    <li><a class="dropdown-item" href="articles/MyocardialInfarction.html">Example: Heart Myocardial Infarcition in Human (Kuppe et. al., 2022)</a></li>
    <li><a class="dropdown-item" href="articles/LR2TF_analysis.html">Intracellular Communication Analysis Bone Marrow Fibrosis in Human</a></li>
    <li><a class="dropdown-item" href="articles/run_liana.html">Ligand-Receptor Inference with Liana</a></li>
    <li><a class="dropdown-item" href="articles/ProgenyLRExample.html">Ligand-Receptor Pathway Enrichment with Progeny - Example</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
<li class="nav-item"><a class="nav-link" href="articles/Install.html">Install</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CostaLab/CrossTalkeR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="crosstalker-">CrossTalkeR <a class="anchor" aria-label="anchor" href="#crosstalker-"></a>
</h1>
</div>
<p><a href="https://doi.org/10.5281/zenodo.4740646" class="external-link"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.4740646.svg" alt="DOI"></a> <a href="https://GitHub.com/CostaLab/CrossTalkeR.js/graphs/commit-activity" class="external-link"><img src="https://img.shields.io/badge/Maintained%3F-yes-green.svg" alt="Maintenance"></a> <a href="https://github.com/CostaLab/CrossTalkeR.js/blob/master/LICENSE" class="external-link"><img src="https://img.shields.io/github/license/CostaLab/CrossTalkeR.svg" alt="GitHub license"></a> <a href="https://GitHub.com/CostaLab/CrossTalkeR.js/releases/" class="external-link"><img src="https://img.shields.io/github/release/CostaLab/CrossTalkeR.svg" alt="GitHub release"></a></p>
<p>James S. Nagai<sup>1</sup>, Nils B. LeimkÃ¼hler<sup>2</sup>, Michael T. Schaub <sup>3</sup>, Rebekka K. Schneider<sup>4,5,6</sup>, Ivan G. Costa<sup>1*</sup></p>
<p><sup>1</sup>Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany</p>
<p><sup>2</sup>Department of Hematology and Stem Cell Transplantation, University Hospital Essen, Germany</p>
<p><sup>3</sup>Department of Computer Science, RWTH Aachen University, Germany</p>
<p><sup>4</sup>Department of Cell Biology, Institute for Biomedical Engineering, Faculty of Medicine,RWTH Aachen University, Pauwelsstrasse 30, 52074 Aachen, NRW, Germany</p>
<p><sup>5</sup>Oncode Institute, Erasmus Medical Center, Rotterdam, 3015GD, the Netherlands</p>
<p><sup>6</sup>Department of Hematology, Erasmus Medical Center, Rotterdam, 3015GD, the Netherlands</p>
<p><img src="reference/figures/CrossTalkeR_only_A.png"></p>
<p><strong>Motivation:</strong> Ligand-receptor (LR) analysis allows the characterization of cellular crosstalk from single cell RNA-seq data. However, current LR methods provide limited approaches for prioritization of cell types, ligands or receptors or characterizing changes in crosstalk between two biological conditions.</p>
<p><strong>Results:</strong> CrossTalkeR is a framework for network analysis and visualisation of LR networks. CrossTalkeR identifies relevant ligands, receptors and cell types contributing to changes in cell communication when contrasting two biological states: disease vs.Â homeostasis. A case study on scRNA-seq of human myeloproliferative neoplasms reinforces the strengths of CrossTalkeR for characterisation of changes in cellular crosstalk in disease state.</p>
<div class="section level2">
<h2 id="install">Install<a class="anchor" aria-label="anchor" href="#install"></a>
</h2>
<p>You can install CrossTalkeR with the simple commands below:</p>
<p><code>{r} install.packages("devtools") devtools::install_github("https://github.com/CostaLab/CrossTalkeR", build_vignettes = TRUE) require(CrossTalkeR)</code></p>
<p><em>Note: Please avoid to use the following characters in celltype name: â€˜$â€™</em></p>
</div>
<div class="section level2">
<h2 id="possible-system-dependencies">Possible system dependencies<a class="anchor" aria-label="anchor" href="#possible-system-dependencies"></a>
</h2>
<pre><code><span><span class="va">libudunits2</span><span class="op">-</span><span class="va">dev</span></span>
<span><span class="va">libgdal</span><span class="op">-</span><span class="va">dev</span></span>
<span><span class="va">gdal</span><span class="op">-</span><span class="va">bin</span></span>
<span><span class="va">libproj</span><span class="op">-</span><span class="va">dev</span></span>
<span><span class="va">proj</span><span class="op">-</span><span class="va">data</span></span>
<span><span class="va">proj</span><span class="op">-</span><span class="va">bin</span></span>
<span><span class="va">libgeos</span><span class="op">-</span><span class="va">dev</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="crosstalker-plots-examples-and-vignette">CrossTalkeR Plots examples and vignette<a class="anchor" aria-label="anchor" href="#crosstalker-plots-examples-and-vignette"></a>
</h2>
<p>We provide in our vignette examples on how to analyse cell interactions from a human myelofibrosis single cell RNA-seq.</p>
<p><code>{r} vignette('CrossTalkeR-HumanMyfib')</code></p>
</div>
<div class="section level2">
<h2 id="integration-of-liana-with-crosstalker">Integration of liana with CrossTalkeR<a class="anchor" aria-label="anchor" href="#integration-of-liana-with-crosstalker"></a>
</h2>
<p>We provide here a tutorial on how to integrate predictions from liana into Crosstalker.</p>
<p><a href="https://github.com/CostaLab/CrossTalkeR/blob/master/vignettes/run_liana.rmd" class="external-link">Run liana</a></p>
</div>
<div class="section level2">
<h2 id="new-features-v140">New Features v1.4.0<a class="anchor" aria-label="anchor" href="#new-features-v140"></a>
</h2>
<ul>
<li>Splitted generate_report function in two parts:
<ul>
<li>analise_LR() to only run the analysis without generating the CrossTalkeR report</li>
<li>make_report() to only generate a new CrossTalkeR report for existing CrossTalkeR results</li>
</ul>
</li>
<li>Added node types to the network:
<ul>
<li>we now consider the annotation of a gene as ligand (L) or receptor (R) to consider the biological function</li>
</ul>
</li>
<li>Less constrains on the cell cluster name annotation (only â€˜$â€™ must be avoided in the cluster naming)</li>
<li>Integration with liana-py for ligand-receptor interaction predictions</li>
</ul>
</div>
<div class="section level2">
<h2 id="old-features-v130">Old Features v1.3.0<a class="anchor" aria-label="anchor" href="#old-features-v130"></a>
</h2>
<ul>
<li>Single and Comparative Reports
<ul>
<li>Cell Cell Interaction visualization</li>
<li>Sending and Receiving Cells Ranking</li>
<li>Gene Target based Sankey Plots</li>
<li>CCI and GCI PCA ranking
<ul>
<li>All measures and PC table</li>
<li>PC1 and PC2 based barplot</li>
</ul>
</li>
<li>LeimkÃ¼hler et. al.Â [2] data were added to the package</li>
<li>Fisher Test were implemented to highlight the CCI edges significance (new) ðŸ”¥<strong>NEW</strong>ðŸ”¥</li>
<li>
<strong>Change input format: Please see the Documentation</strong> ðŸ”¥<strong>NEW</strong>ðŸ”¥
<ul>
<li>A python3 notebook are available to cast the old input to the new input.</li>
</ul>
</li>
<li>Liana (Dimitrov et. al.Â [3]) Output can be used as CrossTalkeR input. ðŸ”¥<strong>NEW</strong>ðŸ”¥</li>
<li>LR pair visualization plot can be done using a Seurat Object ðŸ”¥<strong>NEW</strong>ðŸ”¥</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section level1">
<h1 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h1>
<p>[1] CrossTalkeR: Analysis and Visualisation of Ligand Receptor Networks <a href="https://doi.org/10.1093/bioinformatics/btab370" class="external-link">link</a></p>
<p>[2] Heterogeneous bone-marrow stromal progenitors drive myelofibrosis via a druggable alarmin axis. <a href="https://www.cell.com/cell-stem-cell/fulltext/S1934-5909(20)30542-7#secsectitle0115" class="external-link">link</a></p>
<p>[3] Comparison of Resources and Methods to infer Cell-Cell Communication from Single-cell RNA Data <a href="https://www.biorxiv.org/content/10.1101/2021.05.21.445160v1.full" class="external-link">link</a></p>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/CostaLab/CrossTalkeR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/CostaLab/CrossTalkeR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CrossTalkeR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Vanessa Kloeker <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-7392-9094" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>James Shiniti Nagai <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7655-7206" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Nils LeimkÃ¼hler <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-2530-4329" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Michael T. Schaub <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-2426-6404" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Rebekka Schneider <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-0749-1565" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Ivan Costa Gesteira <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-2890-8697" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vanessa Kloeker, James Shiniti Nagai, Nils LeimkÃ¼hler, Michael T. Schaub, Rebekka Schneider, Ivan Costa Gesteira.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
