<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CrossTalkeR">
<title>CrossTalkeR Cookbook • CrossTalkeR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="CrossTalkeR Cookbook">
<meta property="og:description" content="CrossTalkeR">
<meta property="og:image" content="https://github.com/CostaLab/CrossTalkeR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CrossTalkeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.4.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/CrossTalker_install_basicusage.html">Installation and Usage</a>
    <a class="dropdown-item" href="../articles/HumanFibrosis.html">Human Myelofibrosis example</a>
    <a class="dropdown-item" href="../articles/run_liana.html">liana Tutorial</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="https://github.com/CostaLab/CrossTalkeR">
    <span class="fab fa fab fa fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="CrossTalkeR_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>CrossTalkeR Cookbook</h1>
                        <h4 data-toc-skip class="author">James S. Nagai</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><h4 data-toc-skip class="author">Vanessa Klöker</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><h4 data-toc-skip class="author">Ivan G. Costa</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><small class="dont-index">Source: <a href="https://github.com/CostaLab/CrossTalkeR/blob/HEAD/vignettes/CrossTalkeR.Rmd"><code>vignettes/CrossTalkeR.Rmd</code></a></small>
      <div class="d-none name"><code>CrossTalkeR.Rmd</code></div>
    </address>
</address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="version-info">Version Info<a class="anchor" aria-label="anchor" href="#version-info"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR">CrossTalkeR</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: replacing previous import 'dplyr::as_data_frame' by</span></span>
<span><span class="co">## 'igraph::as_data_frame' when loading 'CrossTalkeR'</span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'dplyr::groups' by 'igraph::groups' when</span></span>
<span><span class="co">## loading 'CrossTalkeR'</span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'clusterProfiler::simplify' by</span></span>
<span><span class="co">## 'igraph::simplify' when loading 'CrossTalkeR'</span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'dplyr::union' by 'igraph::union' when</span></span>
<span><span class="co">## loading 'CrossTalkeR'</span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'igraph::as_data_frame' by</span></span>
<span><span class="co">## 'tibble::as_data_frame' when loading 'CrossTalkeR'</span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'assertthat::has_name' by 'tibble::has_name'</span></span>
<span><span class="co">## when loading 'CrossTalkeR'</span></span></code></pre>
<pre><code><span><span class="co">## Warning: replacing previous import 'igraph::crossing' by 'tidyr::crossing' when</span></span>
<span><span class="co">## loading 'CrossTalkeR'</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p><strong>R version</strong>: R version 4.3.1 (2023-06-16)</p>
<p><strong>Package version</strong>: 1.4.0</p>
</div>
<div class="section level2">
<h2 id="generate-report-example">Generate Report Example<a class="anchor" aria-label="anchor" href="#generate-report-example"></a>
</h2>
<p>In our vignette we provide examples on how to analyse cell interactions from a human myelofibrosis single cell RNA-seq dataset.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'CTR'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span>                               <span class="st">"CTR_LR.csv"</span>,</span>
<span>                               package <span class="op">=</span> <span class="st">"CrossTalkeR"</span><span class="op">)</span>,</span>
<span>           <span class="st">'EXP'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span>                               <span class="st">"EXP_LR.csv"</span>,</span>
<span>                               package <span class="op">=</span> <span class="st">"CrossTalkeR"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'TGFB1|L'</span><span class="op">)</span></span>
<span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"CrossTalkeR"</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_report.html">generate_report</a></span><span class="op">(</span><span class="va">paths</span>,</span>
<span>                        <span class="va">genes</span>,</span>
<span>                        out_path<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">output</span>,<span class="st">'/'</span><span class="op">)</span>,</span>
<span>                        threshold<span class="op">=</span><span class="fl">0</span>,</span>
<span>                        out_file <span class="op">=</span> <span class="st">'vignettes_example.html'</span>,</span>
<span>                        output_fmt <span class="op">=</span> <span class="st">"html_document"</span>,</span>
<span>                        report <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="individual-visualization">Individual Visualization<a class="anchor" aria-label="anchor" href="#individual-visualization"></a>
</h2>
<div class="section level3">
<h3 id="cci">CCI<a class="anchor" aria-label="anchor" href="#cci"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_cci.html">plot_cci</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">CTR</span>,</span>
<span>        colors <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">colors</span>,</span>
<span>        plt_name <span class="op">=</span> <span class="st">'Example 1'</span>,</span>
<span>        coords <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">coords</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">CTR</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,<span class="op">]</span>,</span>
<span>        emax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        leg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        high <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        ignore_alpha <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        log <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        efactor <span class="op">=</span> <span class="fl">8</span>,</span>
<span>        vfactor <span class="op">=</span> <span class="fl">12</span><span class="op">)</span></span></code></pre></div>
<p><img src="CrossTalkeR_files/figure-html/vignete2-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="sankey-plot">Sankey plot<a class="anchor" aria-label="anchor" href="#sankey-plot"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span>lrobj_tbl <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">EXP_x_CTR</span>,</span>
<span>            target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGFB1|L"</span><span class="op">)</span>,</span>
<span>            ligand_cluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>            receptor_cluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>            plt_name <span class="op">=</span> <span class="st">"TGFB1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CrossTalkeR_files/figure-html/vignete4-1.png" width="1440"></p>
</div>
<div class="section level3">
<h3 id="report-samples">Report Samples<a class="anchor" aria-label="anchor" href="#report-samples"></a>
</h3>
<p><a href="Single_vignettes_example.html">Single</a></p>
<p><a href="Comparative_vignettes_example.html">Comparative</a></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Berlin</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] knitr_1.43            colorBlindness_0.1.9  extrafont_0.19       </span></span>
<span><span class="co">##  [4] pals_1.7              graphlayouts_1.0.0    ggrepel_0.9.3        </span></span>
<span><span class="co">##  [7] DT_0.28               ComplexHeatmap_2.16.0 factoextra_1.0.7     </span></span>
<span><span class="co">## [10] patchwork_1.1.2       stringr_1.5.0         tidyr_1.3.0          </span></span>
<span><span class="co">## [13] tibble_3.2.1          dplyr_1.1.2           ggraph_2.1.0         </span></span>
<span><span class="co">## [16] ggplot2_3.4.2         igraph_1.5.0.1        CrossTalkeR_1.4.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.3.1           bitops_1.0-7            ggplotify_0.1.1        </span></span>
<span><span class="co">##   [4] polyclip_1.10-4         lifecycle_1.0.3         rstatix_0.7.2          </span></span>
<span><span class="co">##   [7] doParallel_1.0.17       rprojroot_2.0.3         lattice_0.21-9         </span></span>
<span><span class="co">##  [10] MASS_7.3-60             crosstalk_1.2.0         backports_1.4.1        </span></span>
<span><span class="co">##  [13] magrittr_2.0.3          sass_0.4.7              rmarkdown_2.23         </span></span>
<span><span class="co">##  [16] jquerylib_0.1.4         yaml_2.3.7              mapproj_1.2.11         </span></span>
<span><span class="co">##  [19] cowplot_1.1.1           DBI_1.1.3               RColorBrewer_1.1-3     </span></span>
<span><span class="co">##  [22] maps_3.4.1              abind_1.4-5             zlibbioc_1.46.0        </span></span>
<span><span class="co">##  [25] purrr_1.0.1             BiocGenerics_0.46.0     RCurl_1.98-1.12        </span></span>
<span><span class="co">##  [28] yulab.utils_0.0.6       tweenr_2.0.2            circlize_0.4.15        </span></span>
<span><span class="co">##  [31] GenomeInfoDbData_1.2.10 IRanges_2.34.1          S4Vectors_0.38.1       </span></span>
<span><span class="co">##  [34] enrichplot_1.20.0       tidytree_0.4.4          pkgdown_2.0.7          </span></span>
<span><span class="co">##  [37] codetools_0.2-19        DOSE_3.26.1             ggforce_0.4.1          </span></span>
<span><span class="co">##  [40] tidyselect_1.2.0        shape_1.4.6             aplot_0.1.10           </span></span>
<span><span class="co">##  [43] farver_2.1.1            viridis_0.6.4           matrixStats_1.0.0      </span></span>
<span><span class="co">##  [46] stats4_4.3.1            jsonlite_1.8.7          GetoptLong_1.0.5       </span></span>
<span><span class="co">##  [49] ellipsis_0.3.2          tidygraph_1.2.3         ggalluvial_0.12.5      </span></span>
<span><span class="co">##  [52] iterators_1.0.14        systemfonts_1.0.4       foreach_1.5.2          </span></span>
<span><span class="co">##  [55] tools_4.3.1             treeio_1.24.3           ragg_1.2.5             </span></span>
<span><span class="co">##  [58] sna_2.7-1               Rcpp_1.0.11             glue_1.6.2             </span></span>
<span><span class="co">##  [61] Rttf2pt1_1.3.12         gridExtra_2.3           xfun_0.39              </span></span>
<span><span class="co">##  [64] qvalue_2.32.0           GenomeInfoDb_1.36.1     withr_2.5.0            </span></span>
<span><span class="co">##  [67] fastmap_1.1.1           boot_1.3-28.1           fansi_1.0.4            </span></span>
<span><span class="co">##  [70] SparseM_1.81            digest_0.6.33           networkLite_1.0.5      </span></span>
<span><span class="co">##  [73] R6_2.5.1                mime_0.12               gridGraphics_0.5-1     </span></span>
<span><span class="co">##  [76] textshaping_0.3.6       colorspace_2.1-0        Cairo_1.6-0            </span></span>
<span><span class="co">##  [79] GO.db_3.17.0            dichromat_2.0-0.1       RSQLite_2.3.1          </span></span>
<span><span class="co">##  [82] utf8_1.2.3              generics_0.1.3          data.table_1.14.8      </span></span>
<span><span class="co">##  [85] httr_1.4.6              htmlwidgets_1.6.2       scatterpie_0.2.1       </span></span>
<span><span class="co">##  [88] org.Mm.eg.db_3.17.0     pkgconfig_2.0.3         gtable_0.3.3           </span></span>
<span><span class="co">##  [91] blob_1.2.4              networkDynamic_0.11.3   XVector_0.40.0         </span></span>
<span><span class="co">##  [94] clusterProfiler_4.8.2   shadowtext_0.1.2        htmltools_0.5.5        </span></span>
<span><span class="co">##  [97] carData_3.0-5           fgsea_1.26.0            clue_0.3-64            </span></span>
<span><span class="co">## [100] scales_1.2.1            Biobase_2.60.0          png_0.1-8              </span></span>
<span><span class="co">## [103] ggfun_0.1.1             reshape2_1.4.4          rjson_0.2.21           </span></span>
<span><span class="co">## [106] coda_0.19-4             statnet.common_4.9.0    nlme_3.1-163           </span></span>
<span><span class="co">## [109] org.Hs.eg.db_3.17.0     cachem_1.0.8            GlobalOptions_0.1.2    </span></span>
<span><span class="co">## [112] parallel_4.3.1          HDO.db_0.99.1           AnnotationDbi_1.62.2   </span></span>
<span><span class="co">## [115] desc_1.4.2              pillar_1.9.0            vctrs_0.6.3            </span></span>
<span><span class="co">## [118] ggpubr_0.6.0            car_3.1-2               cluster_2.1.4          </span></span>
<span><span class="co">## [121] extrafontdb_1.0         netdiffuseR_1.22.5      evaluate_0.21          </span></span>
<span><span class="co">## [124] magick_2.7.5            cli_3.6.1               compiler_4.3.1         </span></span>
<span><span class="co">## [127] rlang_1.1.1             crayon_1.5.2            MatchIt_4.5.4          </span></span>
<span><span class="co">## [130] ggsignif_0.6.4          labeling_0.4.2          plyr_1.8.8             </span></span>
<span><span class="co">## [133] fs_1.6.3                stringi_1.7.12          viridisLite_0.4.2      </span></span>
<span><span class="co">## [136] network_1.18.1          BiocParallel_1.34.2     assertthat_0.2.1       </span></span>
<span><span class="co">## [139] munsell_0.5.0           Biostrings_2.68.1       lazyeval_0.2.2         </span></span>
<span><span class="co">## [142] GOSemSim_2.26.1         Matrix_1.5-1            bit64_4.0.5            </span></span>
<span><span class="co">## [145] KEGGREST_1.40.0         highr_0.10              gsw_1.1-1              </span></span>
<span><span class="co">## [148] broom_1.0.5             memoise_2.0.1           oce_1.8-1              </span></span>
<span><span class="co">## [151] bslib_0.5.0             ggtree_3.8.2            fastmatch_1.1-3        </span></span>
<span><span class="co">## [154] bit_4.0.5               downloader_0.4          ape_5.7-1              </span></span>
<span><span class="co">## [157] gson_0.1.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Vanessa Kloeker, James Shiniti Nagai, Nils Leimkühler, Michael T. Schaub, Rebekka Schneider, Ivan Costa Gesteira.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
