<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CrossTalkeR cookbook • CrossTalkeR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CrossTalkeR cookbook">
<meta property="og:description" content="CrossTalkeR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CrossTalkeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CrossTalkeR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/CostaLab/CrossTalkeR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CrossTalkeR cookbook</h1>
                        <h4 data-toc-skip class="author">James S. Nagai</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><h4 data-toc-skip class="author">Nils B. Leimkühler</h4>
            <address class="author_afil">
      Department of Hematology and Stem Cell Transplantation, University Hospital Essen, Germany<br><h4 data-toc-skip class="author">Vanessa Krammer</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><h4 data-toc-skip class="author">Michael T. Schaub</h4>
            <address class="author_afil">
      Department of Computer Science, RWTH Aachen University, Aachen, Germany<br><h4 data-toc-skip class="author">Rebekka K. Schneider</h4>
            <address class="author_afil">
      Department of Hematology, Erasmus Medical Center, Rotterdam, 3015GD, the NetherlandsDepartment of Cell Biology, Institute for Biomedical Engineering, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 GermanyOncode Institute, Erasmus Medical Center, Rotterdam, 3015GD, the Netherlands<br><h4 data-toc-skip class="author">Ivan G. Costa</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><small class="dont-index">Source: <a href="https://github.com/CostaLab/CrossTalkeR/blob/HEAD/../debugCrossTalkeR/vignettes/CrossTalkeR.Rmd" class="external-link"><code>../debugCrossTalkeR/vignettes/CrossTalkeR.Rmd</code></a></small>
      <div class="hidden name"><code>CrossTalkeR.Rmd</code></div>

    </address>
</address>
</address>
</address>
</address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="version-info">Version Info<a class="anchor" aria-label="anchor" href="#version-info"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: replacing previous import 'dplyr::union' by 'igraph::union' when</span>
<span class="co">## loading 'CrossTalkeR'</span></code></pre>
<pre><code><span class="co">## Warning: replacing previous import 'gtools::permute' by 'igraph::permute' when</span>
<span class="co">## loading 'CrossTalkeR'</span></code></pre>
<pre><code><span class="co">## Warning: replacing previous import 'dplyr::as_data_frame' by</span>
<span class="co">## 'igraph::as_data_frame' when loading 'CrossTalkeR'</span></code></pre>
<pre><code><span class="co">## Warning: replacing previous import 'dplyr::groups' by 'igraph::groups' when</span>
<span class="co">## loading 'CrossTalkeR'</span></code></pre>
<pre><code><span class="co">## Warning: replacing previous import 'clusterProfiler::simplify' by</span>
<span class="co">## 'igraph::simplify' when loading 'CrossTalkeR'</span></code></pre>
<pre><code><span class="co">## Warning: replacing previous import 'igraph::as_data_frame' by</span>
<span class="co">## 'tibble::as_data_frame' when loading 'CrossTalkeR'</span></code></pre>
<pre><code><span class="co">## Warning: replacing previous import 'assertthat::has_name' by 'tibble::has_name'</span>
<span class="co">## when loading 'CrossTalkeR'</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span><span class="op">}</span><span class="op">)</span></code></pre></div>
<p><strong>R version</strong>: R version 4.1.3 (2022-03-10)</p>
<p><strong>Package version</strong>: 1.3.2</p>
</div>
<div class="section level2">
<h2 id="generate-report-example">Generate Report Example<a class="anchor" aria-label="anchor" href="#generate-report-example"></a>
</h2>
<p>In our vignette we provide examples on how to analyse cell interactions from a human myelofibrosis single cell RNA-seq dataset.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'CTR'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
                               <span class="st">"ctr_nils_bm_human_newformat.csv"</span>,
                               package <span class="op">=</span> <span class="st">"CrossTalkeR"</span><span class="op">)</span>,
           <span class="st">'EXP'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
                               <span class="st">"exp_nils_bm_human_newformat.csv"</span>,
                               package <span class="op">=</span> <span class="st">"CrossTalkeR"</span><span class="op">)</span><span class="op">)</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'TGFB1'</span><span class="op">)</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_report.html">generate_report</a></span><span class="op">(</span><span class="va">paths</span>,
                        <span class="va">genes</span>,
                        out_path<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">'/html/'</span><span class="op">)</span>,
                        threshold<span class="op">=</span><span class="fl">0</span>,
                        out_file <span class="op">=</span> <span class="st">'vignettes_example.html'</span>,
                        output_fmt <span class="op">=</span> <span class="st">"html_document"</span>,
                        report <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="individual-visualization">Individual Visualization<a class="anchor" aria-label="anchor" href="#individual-visualization"></a>
</h2>
<div class="section level3">
<h3 id="cci">CCI<a class="anchor" aria-label="anchor" href="#cci"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_cci.html">plot_cci</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">CTR</span>,
        colors <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">colors</span>,
        plt_name <span class="op">=</span> <span class="st">'Example 1'</span>,
        coords <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">coords</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">CTR</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,<span class="op">]</span>,
        emax <span class="op">=</span> <span class="cn">NULL</span>,
        leg <span class="op">=</span> <span class="cn">FALSE</span>,
        low <span class="op">=</span> <span class="fl">0</span>,
        high <span class="op">=</span> <span class="fl">0</span>,
        ignore_alpha <span class="op">=</span> <span class="cn">FALSE</span>,
        log <span class="op">=</span> <span class="cn">FALSE</span>,
        efactor <span class="op">=</span> <span class="fl">8</span>,
        vfactor <span class="op">=</span> <span class="fl">12</span><span class="op">)</span></code></pre></div>
<p><img src="CrossTalkeR_files/figure-html/vignete2-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="sankey-plot">Sankey plot<a class="anchor" aria-label="anchor" href="#sankey-plot"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span>lrobj_tbl <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">EXP_x_CTR</span>,
            target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TGFB1"</span><span class="op">)</span>,
            ligand_cluster <span class="op">=</span> <span class="cn">NULL</span>,
            receptor_cluster <span class="op">=</span> <span class="cn">NULL</span>,
            plt_name <span class="op">=</span> <span class="st">"TGFB1"</span><span class="op">)</span></code></pre></div>
<p><img src="CrossTalkeR_files/figure-html/vignete4-1.png" width="1440"></p>
</div>
<div class="section level3">
<h3 id="report-samples">Report Samples<a class="anchor" aria-label="anchor" href="#report-samples"></a>
</h3>
<p><a href="Single_vignettes_example.html">Single</a></p>
<p><a href="Comparative_vignettes_example.html">Comparative</a></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Linux Mint 20.3</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=hu_HU.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=hu_HU.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=hu_HU.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] grid      stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] colorBlindness_0.1.9  extrafont_0.18        pals_1.7             </span>
<span class="co">##  [4] graphlayouts_0.8.0    ggrepel_0.9.1         DT_0.22              </span>
<span class="co">##  [7] ComplexHeatmap_2.10.0 factoextra_1.0.7      patchwork_1.1.1      </span>
<span class="co">## [10] stringr_1.4.0         tidyr_1.2.0           tibble_3.1.7         </span>
<span class="co">## [13] dplyr_1.0.9           ggraph_2.0.5          ggplot2_3.3.6        </span>
<span class="co">## [16] igraph_1.3.1          CrossTalkeR_1.3.2    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] rappdirs_0.3.3         SparseM_1.81           scattermore_0.8       </span>
<span class="co">##   [4] coda_0.19-4            SeuratObject_4.1.0     ragg_1.2.2            </span>
<span class="co">##   [7] bit64_4.0.5            knitr_1.39             irlba_2.3.5           </span>
<span class="co">##  [10] data.table_1.14.2      rpart_4.1.16           KEGGREST_1.34.0       </span>
<span class="co">##  [13] RCurl_1.98-1.6         doParallel_1.0.17      generics_0.1.2        </span>
<span class="co">##  [16] BiocGenerics_0.40.0    org.Mm.eg.db_3.14.0    cowplot_1.1.1         </span>
<span class="co">##  [19] RSQLite_2.2.13         shadowtext_0.1.2       RANN_2.6.1            </span>
<span class="co">##  [22] future_1.25.0          bit_4.0.4              enrichplot_1.14.2     </span>
<span class="co">##  [25] spatstat.data_2.2-0    httpuv_1.6.5           ggsci_2.9             </span>
<span class="co">##  [28] assertthat_0.2.1       viridis_0.6.2          xfun_0.30             </span>
<span class="co">##  [31] Rmagic_2.0.3           jquerylib_0.1.4        evaluate_0.15         </span>
<span class="co">##  [34] promises_1.2.0.1       fansi_1.0.3            DBI_1.1.2             </span>
<span class="co">##  [37] htmlwidgets_1.5.4      spatstat.geom_2.4-0    stats4_4.1.3          </span>
<span class="co">##  [40] purrr_0.3.4            ellipsis_0.3.2         crosstalk_1.2.0       </span>
<span class="co">##  [43] ggpubr_0.4.0           ggnewscale_0.4.7       backports_1.4.1       </span>
<span class="co">##  [46] deldir_1.0-6           vctrs_0.4.1            ggalluvial_0.12.3     </span>
<span class="co">##  [49] Biobase_2.54.0         here_1.0.1             ROCR_1.0-11           </span>
<span class="co">##  [52] abind_1.4-5            cachem_1.0.6           withr_2.5.0           </span>
<span class="co">##  [55] ggforce_0.3.3          progressr_0.10.0       oce_1.7-2             </span>
<span class="co">##  [58] sctransform_0.3.3      sna_2.6                treeio_1.18.1         </span>
<span class="co">##  [61] goftest_1.2-3          cluster_2.1.3          DOSE_3.20.1           </span>
<span class="co">##  [64] ape_5.6-2              lazyeval_0.2.2         crayon_1.5.1          </span>
<span class="co">##  [67] labeling_0.4.2         pkgconfig_2.0.3        MatchIt_4.3.4         </span>
<span class="co">##  [70] tweenr_1.0.2           GenomeInfoDb_1.30.1    nlme_3.1-157          </span>
<span class="co">##  [73] rlang_1.0.2            globals_0.14.0         lifecycle_1.0.1       </span>
<span class="co">##  [76] miniUI_0.1.1.1         gsw_1.0-6              downloader_0.4        </span>
<span class="co">##  [79] extrafontdb_1.0        dichromat_2.0-0.1      rprojroot_2.0.3       </span>
<span class="co">##  [82] polyclip_1.10-0        matrixStats_0.62.0     lmtest_0.9-40         </span>
<span class="co">##  [85] Matrix_1.4-1           aplot_0.1.4            carData_3.0-5         </span>
<span class="co">##  [88] boot_1.3-28            zoo_1.8-10             ggridges_0.5.3        </span>
<span class="co">##  [91] GlobalOptions_0.1.2    png_0.1-7              viridisLite_0.4.0     </span>
<span class="co">##  [94] rjson_0.2.21           bitops_1.0-7           KernSmooth_2.23-20    </span>
<span class="co">##  [97] Biostrings_2.62.0      blob_1.2.3             shape_1.4.6           </span>
<span class="co">## [100] qvalue_2.26.0          parallelly_1.31.1      spatstat.random_2.2-0 </span>
<span class="co">## [103] rstatix_0.7.0          gridGraphics_0.5-1     ggsignif_0.6.3        </span>
<span class="co">## [106] S4Vectors_0.32.4       scales_1.2.0           memoise_2.0.1         </span>
<span class="co">## [109] magrittr_2.0.3         plyr_1.8.7             ica_1.0-2             </span>
<span class="co">## [112] zlibbioc_1.40.0        compiler_4.1.3         scatterpie_0.1.7      </span>
<span class="co">## [115] RColorBrewer_1.1-3     clue_0.3-60            fitdistrplus_1.1-8    </span>
<span class="co">## [118] netdiffuseR_1.22.3     cli_3.3.0              XVector_0.34.0        </span>
<span class="co">## [121] listenv_0.8.0          pbapply_1.5-0          MASS_7.3-56           </span>
<span class="co">## [124] mgcv_1.8-40            tidyselect_1.1.2       stringi_1.7.6         </span>
<span class="co">## [127] textshaping_0.3.6      highr_0.9              yaml_2.3.5            </span>
<span class="co">## [130] GOSemSim_2.20.0        sass_0.4.1             fastmatch_1.1-3       </span>
<span class="co">## [133] tools_4.1.3            future.apply_1.9.0     parallel_4.1.3        </span>
<span class="co">## [136] circlize_0.4.14        rstudioapi_0.13        networkDynamic_0.11.2 </span>
<span class="co">## [139] foreach_1.5.2          gridExtra_2.3          farver_2.1.0          </span>
<span class="co">## [142] Rtsne_0.16             digest_0.6.29          rgeos_0.5-9           </span>
<span class="co">## [145] shiny_1.7.1            Rcpp_1.0.8.3           car_3.0-13            </span>
<span class="co">## [148] broom_0.8.0            later_1.3.0            RcppAnnoy_0.0.19      </span>
<span class="co">## [151] org.Hs.eg.db_3.14.0    httr_1.4.3             AnnotationDbi_1.56.2  </span>
<span class="co">## [154] colorspace_2.0-3       fs_1.5.2               tensor_1.5            </span>
<span class="co">## [157] reticulate_1.24        IRanges_2.28.0         splines_4.1.3         </span>
<span class="co">## [160] uwot_0.1.11            yulab.utils_0.0.4      tidytree_0.3.9        </span>
<span class="co">## [163] spatstat.utils_2.3-1   pkgdown_2.0.3          sp_1.4-7              </span>
<span class="co">## [166] mapproj_1.2.8          ggplotify_0.1.0        plotly_4.10.0         </span>
<span class="co">## [169] systemfonts_1.0.4      xtable_1.8-4           jsonlite_1.8.0        </span>
<span class="co">## [172] ggtree_3.2.1           tidygraph_1.2.1        ggfun_0.0.6           </span>
<span class="co">## [175] R6_2.5.1               pillar_1.7.0           htmltools_0.5.2       </span>
<span class="co">## [178] mime_0.12              glue_1.6.2             fastmap_1.1.0         </span>
<span class="co">## [181] clusterProfiler_4.2.2  BiocParallel_1.28.3    codetools_0.2-18      </span>
<span class="co">## [184] maps_3.4.0             fgsea_1.20.0           utf8_1.2.2            </span>
<span class="co">## [187] lattice_0.20-45        bslib_0.3.1            spatstat.sparse_2.1-1 </span>
<span class="co">## [190] network_1.17.1         leiden_0.3.10          gtools_3.9.2          </span>
<span class="co">## [193] Rttf2pt1_1.3.10        GO.db_3.14.0           survival_3.2-13       </span>
<span class="co">## [196] statnet.common_4.6.0   rmarkdown_2.14         desc_1.4.1            </span>
<span class="co">## [199] munsell_0.5.0          DO.db_2.9              GetoptLong_1.0.5      </span>
<span class="co">## [202] GenomeInfoDbData_1.2.7 iterators_1.0.14       reshape2_1.4.4        </span>
<span class="co">## [205] gtable_0.3.0           spatstat.core_2.4-2    Seurat_4.1.1</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by James Shiniti Nagai, Nils Leimkühler, Vanessa Krammer, Michael T. Schaub, Rebekka Schneider, Ivan Costa Gesteira.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
