<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>CrossTalkeR Example - Human Myelofibrosis • CrossTalkeR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<script src="../extra.js"></script><meta property="og:title" content="CrossTalkeR Example - Human Myelofibrosis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-black" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CrossTalkeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/CrossTalkeR.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="../articles/CrossTalker_basicusage.html">CrossTalkeR - Basic Usage</a></li>
    <li><a class="dropdown-item" href="../articles/run_liana.html">Ligand-Receptor Inference with Liana</a></li>
    <li><a class="dropdown-item" href="../articles/MyocardialInfarction.html">Example: Heart Myocardial Infarction in Human (Kuppe et. al., 2022)</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/Install.html">Install</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CostaLab/CrossTalkeR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>CrossTalkeR Example - Human Myelofibrosis</h1>
                        <h4 data-toc-skip class="author">James S.
Nagai</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074
Germany<br><h4 data-toc-skip class="author">Vanessa
Klöker</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074
Germany<br><h4 data-toc-skip class="author">Ivan
G. Costa</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074 Germany<br><small class="dont-index">Source: <a href="https://github.com/CostaLab/CrossTalkeR/blob/HEAD/vignettes/HumanFibrosis.rmd" class="external-link"><code>vignettes/HumanFibrosis.rmd</code></a></small>
      <div class="d-none name"><code>HumanFibrosis.rmd</code></div>
    </address>
</address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="cell-cell-communication-analysis-of-human-myelofibrosis-scrna-seq-data-using-crosstalker">Cell-Cell Communication Analysis of Human Myelofibrosis scRNA-seq
Data Using CrossTalkeR<a class="anchor" aria-label="anchor" href="#cell-cell-communication-analysis-of-human-myelofibrosis-scrna-seq-data-using-crosstalker"></a>
</h2>
<p>Here, we demonstrate the usage of CrossTalkeR on the case study
Myelofibrosis scRNA-seq data set from the paper. The predicted
ligand-receptor interactions used in this tutorial are provided within
this package. These differ sightly from the results presented in the
paper, since they were recomputed using the cellphoneDB Method with the
liana package and the consensus interaction database.</p>
<p>The scRNA-seq data contains two conditions: control (CTR_LR.csv) and
disease condition (EXP_LR.csv). We first show how to run liana to
retrieve ligand and receptor activities, then how to execute CrossTalkeR
and finally go over some of the results, that are also provided inside
the resulting CrossTalkeR report.</p>
<div class="section level3">
<h3 id="run-liana-for-ligand-receptor-interaction-prediction-in-r">Run liana for Ligand-Receptor Interaction Prediction in R<a class="anchor" aria-label="anchor" href="#run-liana-for-ligand-receptor-interaction-prediction-in-r"></a>
</h3>
<p>For the prediction of ligand-receptor interactions, we are using the
<a href="https://saezlab.github.io/liana/articles/liana_tutorial.html" class="external-link">liana</a>
framework for R. Here we show the usage of liana on our example data and
we are using the CellphoneDB method and the Consensus DB provided with
liana.</p>
<p>First of all, we are laoding all necessary libraries :</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/liana/" class="external-link">liana</a></span><span class="op">)</span></span></code></pre></div>
<p>We assume here that the necessary libraries are already
installed.</p>
<p>Then we can continue to load the Seurat object of our example. The
object is provided in the package as accessible object:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"/path/to/Seurat/object.Rds"</span><span class="op">)</span></span></code></pre></div>
<p>Then we can split the object by sample condition (here meta data
column “protocol”) and run liana for each of the sub-objects, to get
condition specific ligand-receptor interactions. We also filter and
reformat the liana results in the same step, so that the resulting
tables can directly be used as input for the CrossTalkeR analysis:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outpath</span> <span class="op">&lt;-</span> <span class="st">"/path/to/save/results/"</span></span>
<span><span class="va">pval_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span></span>
<span><span class="va">seurat_object_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span><span class="va">seurat_object</span>, split.by <span class="op">=</span> <span class="st">"protocol"</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">condition</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">seurat_object_list</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">sub_object</span> <span class="op">&lt;-</span> <span class="va">seurat_object_list</span><span class="op">[[</span><span class="va">condition</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span>  <span class="va">liana_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/liana/reference/liana_wrap.html" class="external-link">liana_wrap</a></span><span class="op">(</span><span class="va">sub_object</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"cellphonedb"</span>,</span>
<span>    resource <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Consensus"</span><span class="op">)</span>,</span>
<span>    expr_prop <span class="op">=</span> <span class="fl">0.1</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="va">liana_results</span> <span class="op">&lt;-</span> <span class="va">liana_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="va">pval_threshold</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">source</span>, <span class="va">target</span>, <span class="va">ligand</span>, <span class="va">receptor.complex</span>, <span class="va">lr.mean</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>gene_A <span class="op">=</span> <span class="va">ligand</span>, gene_B <span class="op">=</span> <span class="va">receptor.complex</span>, MeanLR <span class="op">=</span> <span class="va">lr.mean</span><span class="op">)</span></span>
<span>  <span class="va">liana_results</span><span class="op">$</span><span class="va">type_gene_A</span> <span class="op">=</span> <span class="st">'Ligand'</span></span>
<span>  <span class="va">liana_results</span><span class="op">$</span><span class="va">type_gene_B</span> <span class="op">=</span> <span class="st">'Receptor'</span></span>
<span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">liana_results</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">outpath</span>, <span class="va">condition</span>, <span class="st">"_LR.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>We have a more general example on how we recommend to run liana in R
<a href="https://costalab.github.io/CrossTalkeR/articles/run_liana.html" class="external-link">here</a>.</p>
</div>
<div class="section level3">
<h3 id="running-crosstalker">Running CrossTalkeR<a class="anchor" aria-label="anchor" href="#running-crosstalker"></a>
</h3>
<p>First, we need to import the CrossTalkeR package and define a named
list of either the paths to the Ligand-Receptor interaction tables, or
directly dataframe objects for each condition of interest. Please note
that it is possible to run CrossTalkeR with more than two conditions.
Here we load the files from within the package. We can further define a
path where we like to save our results (output) and run CrossTalkeR with
the generate_report() function:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">CTR</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">EXP</span><span class="op">)</span></span>
<span><span class="va">paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"CTR"</span> <span class="op">=</span> <span class="va">CTR</span>,</span>
<span>              <span class="st">"EXP"</span> <span class="op">=</span> <span class="va">EXP</span><span class="op">)</span></span>
<span></span>
<span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"CrossTalkeR"</span><span class="op">)</span></span>
<span><span class="va">ctkroutput</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_report.html">generate_report</a></span><span class="op">(</span><span class="va">paths</span>,</span>
<span>                              out_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">output</span>, <span class="st">"/"</span><span class="op">)</span>,</span>
<span>                              out_file <span class="op">=</span> <span class="st">"vignettes_example.html"</span>,</span>
<span>                              output_fmt <span class="op">=</span> <span class="st">"html_document"</span>,</span>
<span>                              report <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                              filtered_net <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Besides passing the paths to the interaction tables and the output
folder, we here also define: out_file - A suffix for the result reports
output_fmt - The file type of the result reports report - A boolean
value if the reports should be created at all</p>
<p>As result of the successful execution, we receive three seurat
objects (LR_data.Rds, LR_data_step2.Rds and LR_data_final.Rds), as well
as two html-reports(SingleCondition.html and ComparativeCondition.html).
The LR_data_final.Rds contains all results produced by CrossTalkeR and
the two html-reports contain the visualization of the results inside the
R object. It is possible to load the results of the CrossTalkeR analysis
from the “LR_data_final.Rds” R-object and produce the plots separately.
In the following we are going to take a look at different results within
the two reports, step by step.</p>
</div>
<div class="section level3">
<h3 id="single-condition-report">Single Condition Report<a class="anchor" aria-label="anchor" href="#single-condition-report"></a>
</h3>
<p>The first report contains the results for each analysed condition
considered separately. In our case in each section is a plot for the CTR
condition (CTR) and the disease condition (EXP). There are two main
sections in the report. The first one only deals with
cell-cell-interactions and the second with cell-gene-interactions, which
here means an analysis at the ligand and receptor gene level.</p>
<div class="section level4">
<h4 id="cell-cell-interaction-analysis-results">Cell-Cell-Interaction Analysis Results<a class="anchor" aria-label="anchor" href="#cell-cell-interaction-analysis-results"></a>
</h4>
<p>The cell-cell-interaction (CCI) graph plots focus on the interactions
between the different cell types within the data. We focus on three
measures inside this plot:</p>
<ol style="list-style-type: decimal">
<li>Percentage of interactions indicated by the opacity of the
arrows</li>
<li>Edge weight (sum of single L-R interaction scores between cell type
pairs) indicated by the edge color</li>
<li>Pagerank (Node importance) indicated by the node size</li>
</ol>
<p>After loading the result data from the “LR_data_final.Rds”, we can
plot the cell-cell interactions with the plot_cci() function. Here we
show a plot for each analysed condition:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ctkroutput</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">ctkroutput</span> </span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_cci.html">plot_cci</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">CTR</span>,</span>
<span>        colors <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">colors</span>,</span>
<span>        plt_name <span class="op">=</span> <span class="st">'Control'</span>,</span>
<span>        coords <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">coords</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">CTR</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,<span class="op">]</span>,</span>
<span>        emax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        leg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        high <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        ignore_alpha <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        log <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        efactor <span class="op">=</span> <span class="fl">8</span>,</span>
<span>        vfactor <span class="op">=</span> <span class="fl">12</span>,</span>
<span>        pg <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">[[</span><span class="st">"CTR"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">Pagerank</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_cci.html">plot_cci</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">EXP</span>,</span>
<span>        colors <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">colors</span>,</span>
<span>        plt_name <span class="op">=</span> <span class="st">'Disease'</span>,</span>
<span>        coords <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">coords</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">EXP</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,<span class="op">]</span>,</span>
<span>        emax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        leg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        high <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        ignore_alpha <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        log <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        efactor <span class="op">=</span> <span class="fl">8</span>,</span>
<span>        vfactor <span class="op">=</span> <span class="fl">12</span>,</span>
<span>        pg <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">[[</span><span class="st">"EXP"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">Pagerank</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p1</span><span class="op">+</span><span class="va">p2</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete2-1.png" width="960"><img src="HumanFibrosis_files/figure-html/vignete2-2.png" width="960"></p>
<p>In the CTR plot, we can see that a high proportion of interactions
involve MSCs communicating with themselves, as well as MSCs and Myeloid
cells interacting with each other. These interactions also have a higher
weight, as indicated by the edge color. The most important nodes
according to Pagerank are the MSC, Myeloid, and Megakaryocyte nodes.
When analyzing the plot of the EXP condition, we again observe a high
proportion of interactions involving self-communicating MSCs, and one of
the most significant nodes by pagerank remains the MSC node. Here, the
MSCs are predicted to communicate more with Megakaryocytes and
Fibroblasts than with Myeloid cells. To effectively compare the two
conditions, we recommend utilizing the comparative report results.</p>
</div>
<div class="section level4">
<h4 id="analysis-on-the-gene-cell-interaction-level">Analysis on the Gene-Cell Interaction Level<a class="anchor" aria-label="anchor" href="#analysis-on-the-gene-cell-interaction-level"></a>
</h4>
<p>We provide several different possibilities to analyse the
ligand-receptor interactions on the gene level. In CrossTalkeR,
topological network measures are calculated for each node(gene) in the
build networks. These measures include Influencer (depending on outgoing
edges), Listener(depending on incoming edges), Mediator (depending on
outgoing and incoming edges), and Pagerank (overall importance)
rankings. The ligand and receptors can be ranked by these measures and
we can identify possibly relevant genes by considering the top genes of
the rankings. In the report we provide an interactive table with all the
rankings for each gene in the network. Here is an example for the top 10
genes in the Pagerank ranking for both preset conditions in form of
Barplots:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rankings_table</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">$</span><span class="va">CTR_ggi</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Pagerank</span><span class="op">)</span></span>
<span><span class="va">rankings_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">rankings_table</span>, n<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">signal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">rankings_table</span><span class="op">$</span><span class="va">Pagerank</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">'negative'</span>, <span class="st">'positive'</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">rankings_table</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Pagerank</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html" class="external-link">reorder</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">Pagerank</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">signal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Blue2DarkOrange18Steps</span><span class="op">[</span><span class="fl">14</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Top Listener in Control Condition"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">rankings_table</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">$</span><span class="va">EXP_ggi</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Pagerank</span><span class="op">)</span></span>
<span><span class="va">rankings_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">rankings_table</span>, n<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">signal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">rankings_table</span><span class="op">$</span><span class="va">Pagerank</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">'negative'</span>, <span class="st">'positive'</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">rankings_table</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Pagerank</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html" class="external-link">reorder</a></span><span class="op">(</span><span class="va">nodes</span>, <span class="va">Pagerank</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">signal</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Blue2DarkOrange18Steps</span><span class="op">[</span><span class="fl">14</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Top Listener in Disease Condition"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p1</span><span class="op">+</span><span class="va">p2</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete3-1.png" width="960"></p>
<p>It is possible to perform further analysis on the gene-gene-level
regarding the other topological measures, but we focus on the result of
these more in the condition comparison results described in the next
section.</p>
</div>
</div>
<div class="section level3">
<h3 id="compared-condition-results">Compared Condition Results<a class="anchor" aria-label="anchor" href="#compared-condition-results"></a>
</h3>
<p>CrossTalkeR not only analyzes the conditions individually but also
calculates the differences in the cell-cell interactions between the
conditions. The results of these analyses are presented in the
comparative report. In our example we compare the EXP against the CTR
condition.</p>
<div class="section level4">
<h4 id="cell-cell-interaction-analysis-results-1">Cell-Cell-Interaction analysis results<a class="anchor" aria-label="anchor" href="#cell-cell-interaction-analysis-results-1"></a>
</h4>
<p>As before for the single condition results, the first part of the
results deal with cell-cell interactions. Here we show again the CCI
graph plot, but now for the compared conditions:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_cci.html">plot_cci</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">EXP_x_CTR</span>,</span>
<span>        colors <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">colors</span>,</span>
<span>        plt_name <span class="op">=</span> <span class="st">'Disease vs Control'</span>,</span>
<span>        coords <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">coords</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">EXP_x_CTR</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,<span class="op">]</span>,</span>
<span>        emax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        leg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        high <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        ignore_alpha <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        log <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        efactor <span class="op">=</span> <span class="fl">8</span>,</span>
<span>        vfactor <span class="op">=</span> <span class="fl">12</span>, pg <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">[[</span><span class="st">"EXP_x_CTR"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">Pagerank</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete4-1.png" width="960">
The main difference from the previous CCI graph plots is that the edge
weights can now be not only positive but also negative, as well as the
Pagerank values of the nodes. Positive values mean that these
interactions between cell types have a higher activity/importance in the
EXP condition, while negative values indicate a lower
activity/importannce in the EXP condition (higher in CTR). As we can see
in the CCI plot above, there are equally positive and negative edge
weights present. The edge weights between MSCs and Megakaryocytes are
positive up in both directions in the Disease condition. Thus, we can
conclude that the communication is stronger in the EXP example. In
contrast, the edge weights between the MSC and Myeloid nodes have
negative edge weights, meaning these communication are lower in the EXP
condition. Further, the self communication edge of the MSC node has a
strong negative weight and also the pagerank is very small (indicated by
small node size). This indicates that MSCs are more important in the
CCIs in the CTR condition.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">EnhancedVolcano</span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">stats</span><span class="op">$</span><span class="va">EXP_x_CTR</span>, lab<span class="op">=</span><span class="va">data</span><span class="op">@</span><span class="va">stats</span><span class="op">$</span><span class="va">EXP_x_CTR</span><span class="op">$</span><span class="va">columns_name</span>,x<span class="op">=</span><span class="st">'lodds'</span>,y<span class="op">=</span><span class="st">'p'</span>,pCutoff<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete451-1.png" width="960"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_cci.html">plot_cci</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">EXP_x_CTR_filtered</span>,</span>
<span>        colors <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">colors</span>,</span>
<span>        plt_name <span class="op">=</span> <span class="st">'Disease vs Control Fisher Test Filtered'</span>,</span>
<span>        coords <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">coords</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">EXP_x_CTR_filtered</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,<span class="op">]</span>,</span>
<span>        emax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        leg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        high <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        ignore_alpha <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        log <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        efactor <span class="op">=</span> <span class="fl">8</span>,</span>
<span>        vfactor <span class="op">=</span> <span class="fl">12</span>, pg <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">[[</span><span class="st">"EXP_x_CTR_filtered"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">Pagerank</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete452-1.png" width="960">
CrossTalkeR offers a refined CCI graph by employing Fisher’s exact test
to filter the network based on interaction proportions. In this small
example, the interactions with Fibroblasts and Neural cells are removed
from the network applying the tests results. The before described
interactions between MSCs, Megakaryocytes and Myeloid cells are still
retained, further indicating the importance in the communication. Since
this example data is small, we going to continue using the unfiltered
network in the following steps. The filtered CCI graph is mostly helpful
for large communication networks.</p>
<p>Since we have created a CCI network we can also analyze it based on
topological properties to identify possible biologically relevant
processes. For this purpose, the same topological measures can be used
that are present in the single condition results for gene-cell
interactions analysis (Influencer, Listener, Mediator, Pagerank). As
before, we can look at the rankings individually in the form of bar
plots. Here is an example for the Pagerank of cell types in the
network:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"EXP_x_CTR"</span>, <span class="st">"Pagerank"</span>, type <span class="op">=</span> <span class="cn">NULL</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete5-1.png" width="960"></p>
<p>In our comparison, the Megakaryocytes have the highest pagerank,
which suggest a high importance in our network for the EXP condition. In
the negative direction, Myeloid cells has the lowest Pagerank. This
suggest a higher importance in the CTR condition. By considering now
also the other topological measures, we can assign/find other
communication properties of the cell types:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"EXP_x_CTR"</span>, <span class="st">"Influencer"</span>, type <span class="op">=</span> <span class="cn">NULL</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete7_Influencer-1.png" width="960"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"EXP_x_CTR"</span>, <span class="st">"Listener"</span>, type <span class="op">=</span> <span class="cn">NULL</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete7_Influencer-2.png" width="960"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"EXP_x_CTR"</span>, <span class="st">"Mediator"</span>, type <span class="op">=</span> <span class="cn">NULL</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete7_Influencer-3.png" width="960"></p>
<p>Including the Influencer, Listener and Mediator barplots reveals that
Megakaryocytes have the highest PageRank, Influencer, and Listener
scores, indicating they engage in the most significant interactions in
the EXP condition. However, their Mediator score is negative, suggesting
a stronger mediating role in the CTR condition.</p>
<p>In contrast, Myeloid cells show high negative scores across most
metrics except for the Influencer score, emphasizing their importance in
the CTR condition. These findings highlight the differing roles of these
cell types in EXP versus normal states.</p>
</div>
<div class="section level4">
<h4 id="cell-gene-interaction-analysis-results">Cell-Gene-Interaction Analysis Results<a class="anchor" aria-label="anchor" href="#cell-gene-interaction-analysis-results"></a>
</h4>
<p>The second part of the report deals again with the interactions on
the gene level. Here, the procedure is similar to the one described in
the last section for the cell-cell interactions and the analysis
performed for the single conditions. First of all, we can look at the
results of the topological rankings in the form of bar plots, again
using the example of the pagerank:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"EXP_x_CTR_ggi"</span>, <span class="st">"Pagerank"</span>, type <span class="op">=</span> <span class="cn">NULL</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span>, mode <span class="op">=</span> <span class="st">"cgi"</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete9-1.png" width="960"></p>
<p>The plot highlights the ligand and receptor genes that play
significant roles in different conditions. Notably, the top genes in
both positive and negative directions are predominantly receptor genes.
Specifically, the majority of positively scored genes are receptor genes
from Megakaryocytes, while the top negatively scored genes are
associated with Myeloid cells.</p>
<p>Among the key receptors, ITGB1 is upregulated in Megakaryocytes,
while ITGA1 is downregulated in MSCs. Additionally, the score of CXCR4
is also elevated in Megakaryocytes. For further insights, we encourage
examining the other rankings. Given that we are focusing on pairwise
interactions, our analysis will concentrate on the Influencer and
Listener rankings:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"EXP_x_CTR_ggi"</span>, <span class="st">"Influencer"</span>, type <span class="op">=</span> <span class="st">"L"</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span>, mode <span class="op">=</span> <span class="st">"cgi"</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete9_other-1.png" width="960"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"EXP_x_CTR_ggi"</span>, <span class="st">"Listener"</span>, type <span class="op">=</span> <span class="st">"R"</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span>, mode <span class="op">=</span> <span class="st">"cgi"</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/vignete9_other-2.png" width="960"></p>
<p>The top Influencer in EXP condition identified is TGFB1 in
Megakaryocytes. Also COL1A1 and COL1A2 from MSCs rank high on the
positive side. CXCL12 from MSCs as example has a negative Influencer
ranking. In the Listener category, integrin receptors dominate, but
there are also receptors as CXCR4 and LRP1 ranking high/low.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="follow-up-analysis">Follow-up Analysis<a class="anchor" aria-label="anchor" href="#follow-up-analysis"></a>
</h2>
<p>Since we got a first impression of the data with the previous
results, we can further deepen the analysis. For example, we had TGFB1
in Megakaryocytes, ITGB1 in Megakaryocytes, Fibroblasts and MSCs and
CXCR4 in Megakaryocytes and Myeloid cells as interesting genes. With
CrossTalkeR, we can look even more closely at the interactions involving
these genes using Sankey plots. There are two ways to do this. First, we
can regenerate the report again, this time passing a list of genes of
interest:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'TGFB1|L'</span>,</span>
<span>                       <span class="st">'ITGB1|R'</span>,</span>
<span>                       <span class="st">'CXCR4|R'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_report.html">make_report</a></span><span class="op">(</span>out_path <span class="op">=</span> <span class="va">output_path</span>,</span>
<span>                    genes <span class="op">=</span> <span class="va">genes_of_interest</span>,</span>
<span>                    out_file <span class="op">=</span> <span class="st">'vignettes_example.html'</span>,</span>
<span>                    output_fmt <span class="op">=</span> <span class="st">"html_document"</span>,</span>
<span>                    report <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    org <span class="op">=</span> <span class="st">"hsa"</span><span class="op">)</span></span></code></pre></div>
<p>The second option is to plot only the Sankey plots for the genes of
interests, without redoing the report:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">EXP_x_CTR</span>,</span>
<span>            target <span class="op">=</span> <span class="st">"TGFB1|L"</span>,</span>
<span>            ligand_cluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>            receptor_cluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>            plt_name <span class="op">=</span> <span class="st">"TGFB1 ligand gene interactions EXP vs CTR"</span>,</span>
<span>            threshold <span class="op">=</span> <span class="fl">50</span>, tfflag <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/unnamed-chunk-7-1.png" width="1056"></p>
<p>In the Sankey plot, not all interactions involving the TGFB1 ligand
gene are shown, but only those with a high score. The number of actual
interactions can be controlled with the threshold parameter. In the
example above 50 interactions are selected. Since we have only entered
one target gene without any further information, we can now also see
interactions originating from different cell types, here Fibroblasts,
Megakaryocytes and MSCs. Furthermore, we now have a list of receptors
with associated cell types with which the TGFB1 ligand can possibly
interact. The list also includes the receptor ITGB1 in Megakaryocytes,
which was also a gene of interest. The interaction between TGFB1 and
ITGB1 has a positive score, which means it is enriched in the EXP
condition.</p>
<p>The third gene of interest was the receptor gene CXCR4 with a high
importance in Megakaryocytes in the EXP condition and in Myeloid cells
in the CTR condition. If we want to focus only on CXCR4 in
Megakaryocytes and Myeloid cells, we can define a receptor_cluster value
when running the Sankey plot function. Also, the number of interactions
included is reduced by setting the threshold parameter to 15.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">EXP_x_CTR</span>,</span>
<span>            target <span class="op">=</span> <span class="st">"CXCR4|R"</span>,</span>
<span>            ligand_cluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>            receptor_cluster <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Megakaryocyte"</span>, <span class="st">"Myeloid"</span><span class="op">)</span>,</span>
<span>            plt_name <span class="op">=</span> <span class="st">"Megakaryocyte and  Myeloid CXCR4 receptor gene interaction EXP vs CTR"</span>,</span>
<span>            threshold <span class="op">=</span> <span class="fl">15</span>, tfflag <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/unnamed-chunk-8-1.png" width="1056"></p>
<p>The Sankey plot confirms that the LR Score indicates higher
interaction with CXCR4 in Megakaryocytes under EXP condition and in
Myeloid cells under CTR condition. Additionally, it highlights
previously occuring ligand and receptors, including TGFB ligand from
Megakaryocytes to CXCR4 in Megakaryocytes and CXCL12 from MSCs to CXCR4
in both Megakaryocytes and Myeloid cells.</p>
<p>Another interesting target described in by <a href="https://www.sciencedirect.com/science/article/pii/S1934590920305427?via%3Dihub" class="external-link">Leimkühler
et. al. 2021</a> is the ligand S100A8. The study also shows that when
S100A8 is targeted the MPN phenotype is ameliorated</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">EXP_x_CTR</span>,</span>
<span>            target <span class="op">=</span> <span class="st">"S100A8|L"</span>,</span>
<span>            ligand_cluster <span class="op">=</span> <span class="st">"Myeloid"</span>,</span>
<span>            receptor_cluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>            plt_name <span class="op">=</span> <span class="st">"Myeloid S100A8 receptor gene interaction EXP vs CTR"</span>,</span>
<span>            threshold <span class="op">=</span> <span class="fl">15</span>, tfflag <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="HumanFibrosis_files/figure-html/unnamed-chunk-9-1.png" width="1056"></p>
</div>
<div class="section level2">
<h2 id="report-samples">Report Samples<a class="anchor" aria-label="anchor" href="#report-samples"></a>
</h2>
<p><a href="https://github.com/CostaLab/CrossTalkeR/blob/master/inst/extdata/Single_vignettes_example.html" class="external-link">Single
Condition Report</a></p>
<p><a href="https://github.com/CostaLab/CrossTalkeR/blob/master/inst/extdata/Comparative_vignettes_example.html" class="external-link">Comparative
Condition Report</a></p>
</div>
<div class="section level2">
<h2 id="further-tutorials">Further Tutorials<a class="anchor" aria-label="anchor" href="#further-tutorials"></a>
</h2>
<p>If you want to learn more about how to infer ligand and receptor
interactions with liana and how to use it with CrossTalkeR, check out
this tutorial: <a href="https://costalab.github.io/CrossTalkeR/articles/run_liana.html" class="external-link">Run
liana</a>.</p>
<p><br><br><br><br></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vanessa Kloeker, James Shiniti Nagai, Nils Leimkühler, Michael T. Schaub, Rebekka Schneider, Ivan Costa Gesteira.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>



  pkgdown/extra.js

  </body>
</html>
