<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Intracellular Communication Analysis • CrossTalkeR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Intracellular Communication Analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-black" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CrossTalkeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/CrossTalkeR.html">Get started</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="../articles/CrossTalker_basicusage.html">CrossTalkeR 101</a></li>
    <li><a class="dropdown-item" href="../articles/HumanFibrosis.html">Example: Bone Marrow Fibrosis in Human (Leimkuhler et. al., 2021)</a></li>
    <li><a class="dropdown-item" href="../articles/MyocardialInfarction.html">Example: Heart Myocardial Infarcition in Human (Kuppe et. al., 2022)</a></li>
    <li><a class="dropdown-item" href="../articles/LR2TF_analysis.html">Intracellular Communication Analysis Bone Marrow Fibrosis in Human</a></li>
    <li><a class="dropdown-item" href="../articles/run_liana.html">Ligand-Receptor Inference with Liana</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/Install.html">Install</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CostaLab/CrossTalkeR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Intracellular Communication Analysis</h1>
                        <h4 data-toc-skip class="author">Vanessa
Klöker</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074
Germany<br><h4 data-toc-skip class="author">James
S. Nagai</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074
Germany<br><h4 data-toc-skip class="author">Ivan
G. Costa</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074 Germany<br><small class="dont-index">Source: <a href="https://github.com/CostaLab/CrossTalkeR/blob/HEAD/vignettes/LR2TF_analysis.rmd" class="external-link"><code>vignettes/LR2TF_analysis.rmd</code></a></small>
      <div class="d-none name"><code>LR2TF_analysis.rmd</code></div>
    </address>
</address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="intracellular-communication-analysis-on-human-myelofibrosis-scrna-seq-data-using-the-lr2tf-package-and-crosstalker">Intracellular Communication Analysis on Human Myelofibrosis
scRNA-seq Data Using the LR2TF Package and CrossTalkeR<a class="anchor" aria-label="anchor" href="#intracellular-communication-analysis-on-human-myelofibrosis-scrna-seq-data-using-the-lr2tf-package-and-crosstalker"></a>
</h2>
<p>This is a guide for the usage of the LR2TF package to infer TF
activities from scRNA-seq data. The transcription factor activities are
estimated using the decoupleR tool. Using the DoRothEA regulon version
from decoupleR and post-translational interactions from the Omnipath
database[2,3,4], connections are made between transcription factors and
ligands and receptors.</p>
<p>The results can be combined with ligand-receptor interactions and
then analyzed using CrossTalkeR (<a href="https://github.com/CostaLab/CrossTalkeR/" class="external-link uri">https://github.com/CostaLab/CrossTalkeR/</a>) [5].</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>The package can be installed directly from github:  </p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"CostaLab/LR2TF"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"CostaLab/LR2TF"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p> </p>
</div>
<div class="section level3">
<h3 id="transcription-factor-activity-predictions-using-decoupler-in-python">Transcription Factor Activity Predictions Using decoupleR in
Python<a class="anchor" aria-label="anchor" href="#transcription-factor-activity-predictions-using-decoupler-in-python"></a>
</h3>
<p>For the transcription factor activity predictions we recommend using
the decoupleR package in its python version. This provides a better
performance, especially for larger data sets. To perform the predictions
with decoupleR, we need an AnnData (.h5ad) file. If the scRNA-seq data
is saved as a Seurat object (.Rds file), it can be converted to AnnData
format using the sceasy package and the function we provide:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">LR2TF</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load own dataset:</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"/path/to/seurat_object.Rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># test dataset from package:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">bone_marrow_stromal_cell_example</span>, package <span class="op">=</span> <span class="st">"LR2TF"</span><span class="op">)</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="va">bone_marrow_stromal_cell_example</span></span>
<span></span>
<span><span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/convert_seurat_to_anndata.html" class="external-link">convert_seurat_to_anndata</a></span><span class="op">(</span><span class="va">seurat_object</span>, <span class="st">"/path/to/save/AnnData/object/"</span><span class="op">)</span></span></code></pre></div>
<p>The file “anndata_object.h5ad” will be saved into the user defined
path and then used to perform the predictions. Beside the scRNA-seq data
file, we also need to define a regulon database in form of a csv file
with the coloumns “source”, “target” and “weight”. Within this package
we provide the dorothea databases for human and mouse, downloaded from
the decoupleR package. These files also contain the column “confidence”
(levels A to D) with information on how well described a transcription
factor and target gene interaction is in different resources. We
recommend using the confidence levels A and B.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">regulon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"regulons"</span>, <span class="st">"human_dorothea_reg.csv"</span>, package <span class="op">=</span> <span class="st">"LR2TF"</span><span class="op">)</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">filtered_regulon</span> <span class="op">&lt;-</span> <span class="va">regulon</span><span class="op">[</span><span class="va">regulon</span><span class="op">$</span><span class="va">confidence</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="st">"/path/to/save/filterd_regulon.csv"</span><span class="op">)</span></span></code></pre></div>
<p>Now it is possible to run the transcription factor activity
prediction with decoupleR. Here we are using the Univariate Linear Model
(ulm) approach recommended by the authors:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="im">import</span> decoupler <span class="im">as</span> dc</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>ann_data <span class="op">=</span> sc.read_h5ad(<span class="st">"/path/to/AnnData/object/anndata_object.h5ad"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>reg <span class="op">=</span> pd.read_csv(<span class="st">"/path/to/filterd_regulon.csv"</span>)</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>dc.run_ulm(mat<span class="op">=</span>ann_data, net<span class="op">=</span>reg, source<span class="op">=</span><span class="st">'source'</span>, target<span class="op">=</span><span class="st">'target'</span>, weight<span class="op">=</span><span class="st">'weight'</span>, verbose<span class="op">=</span><span class="va">True</span>, use_raw<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>estimates <span class="op">=</span> ann_data.obsm[<span class="st">'ulm_estimate'</span>]</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>estimates.to_csv(<span class="st">"/path/to/save/decoupler_results.csv"</span>)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="using-the-lr2tf-package">Using the LR2TF package<a class="anchor" aria-label="anchor" href="#using-the-lr2tf-package"></a>
</h3>
<p>Now, that we have a transcription factor activity matrix, we can
continue the analysis with the LR2TF package. In this case we will use
our test data for this example. First of all, it is necessary to define
the following parameters in form of a list:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parameters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"out_path"</span> <span class="op">=</span> <span class="st">"/path/to/save/results"</span>,</span>
<span>  reg <span class="op">=</span> <span class="st">"/path/to/filterd_regulon.csv"</span>,</span>
<span>  <span class="st">"organism"</span> <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>  <span class="st">"celltype"</span> <span class="op">=</span> <span class="st">"new_annotation"</span>, <span class="co"># name of the meta data field defining cell identities</span></span>
<span>  <span class="st">"condition"</span> <span class="op">=</span> <span class="st">"protocol"</span>, <span class="co"># name of the meta data field defining conditions</span></span>
<span>  <span class="st">"comparison_list"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PMF,MF2"</span>, <span class="st">"control"</span><span class="op">)</span><span class="op">)</span>, <span class="co"># list of condition comparison to consider</span></span>
<span>  <span class="st">"logfc"</span> <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  <span class="st">"pval"</span> <span class="op">=</span> <span class="fl">0.05</span>, <span class="co"># thresholds for logfc and pval used in differential transcription factor analysis</span></span>
<span>  <span class="st">"num_cell_filter"</span> <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span> <span class="co"># define the minimum number of cells per cluster to perform the analysis on</span></span></code></pre></div>
<p>After defining the necessary parameter the transcription factor
activity can be performed by calling:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu">tf_activity_analysis</span><span class="op">(</span></span>
<span>  seuratobject <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>  tf_activities <span class="op">=</span> <span class="st">"/path/to/decoupler_results.csv"</span>,</span>
<span>  arguments_list <span class="op">=</span> <span class="va">parameters</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The “results” object contains the results of the performed analyses,
consisting of multiple tables inside the object:</p>
<ol style="list-style-type: decimal">
<li>
<strong>tf_activities_condition:</strong> Tables with condition
significant transcription factors for each compared condition</li>
<li>
<strong>tf_activities_cluster:</strong> Tables with cluster specific
transcription factors for all conditions in the data</li>
<li>
<strong>average_gene_expression:</strong> Matrices for each
condition with average gene expressions</li>
<li>
<strong>regulon:</strong> Regulon used for the analysis as specified
by the user</li>
<li>
<strong>CTR_input_condition:</strong> for each condition a table
with receptor-transcription factor and transcription factor-ligand
interactions based on condition specific transcription factors (input
for CrossTalker)</li>
<li>
<strong>CTR_input_cluster:</strong> for each condition a table with
receptor-transcription factor and transcription factor-ligand
interactions based on cluster specific transcription factors (input for
CrossTalker)</li>
<li>
<strong>intracellular_network_condition:</strong> for each condition
a table with receptor-transcription factor and transcription
factor-target gene interactions based on condition specific
transcription factors</li>
<li>
<strong>intracellular_network_cluster:</strong> for each condition a
table with receptor-transcription factor and transcription factor-target
gene interactions based on cluster specific transcription factors</li>
</ol>
<p>(Note that special characters might be exchanged by underscores, if
they cause problems with the naming of the tables; eg PMF,MF2 -&gt;
PMF_MF2)</p>
<div class="section level4">
<h4 id="transcription-factor-activity-analysis-plots">Transcription Factor Activity Analysis Plots<a class="anchor" aria-label="anchor" href="#transcription-factor-activity-analysis-plots"></a>
</h4>
<p>As already outlined in the result object description, two different
analysis approaches are used to identify which predicted TF activities
are significant. The first approach focuses on the comparison between
different sample conditions (here, control vs. fibrosis). During the
analysis, a large heatmap is generated that includes all TFs showing
significant differential activity between the conditions in at least one
of the cell clusters.</p>
<img style="display: block; margin: auto; height: 100%; width: 100%;" src="TF_images%2FTF_large_heatmap.png"><figcaption>
Heatmap showing TFs that show significant differences in their
activities between the PMF_MF2 and control condition. The significance
level is given by the asterisk symbol: * p-value &lt; 0.05, ** p-value
&lt; 0.01, *** p-value &lt; 0.001.
</figcaption><p> </p>
<p>The second approach focuses on identifying cluster-specific TFs
within the different conditions separately. In addition to a similar
heatmap as in the condition comparison, heatmaps are provided to show
the most variable TFs among all cell clusters.</p>
<div style="display: flex; align-items: center; justify-content: center;">
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="TF_images%2FControl_Variable_Gene_Heatmap.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;">
Heatmap showing TFs with highly variable activity between the clusters
in the control condition.
</figcaption></figure><figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="TF_images%2FPMF_MF2_Variable_Gene_Heatmap.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;">
Heatmap showing TFs with highly variable activity between the clusters
in the PMF_MF2 condition.
</figcaption></figure>
</div>
<p> </p>
</div>
</div>
<div class="section level3">
<h3 id="running-crosstalker-with-tf-predictions">Running CrossTalkeR with TF Predictions<a class="anchor" aria-label="anchor" href="#running-crosstalker-with-tf-predictions"></a>
</h3>
<p>The last step is to combine previous results from ligand receptor
interaction analyses (e.g. CellPhoneDB) with the transcription factor
results. (In the case of the test data the ligand-receptor interactions
are provided within the CrossTalkeR package. Note that these results
include receptor complexes.)</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># First possibility: Read the csv files with LR interaction results and then combine them with the TF interaction results</span></span>
<span><span class="va">table_ctr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"/path/to/control_lr_results.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">table_exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"/path/to/PMF,MF2_lr_results.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ctr_inptu</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF_complexes.html" class="external-link">combine_LR_and_TF_complexes</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">[[</span><span class="st">"control"</span><span class="op">]</span><span class="op">]</span>, <span class="va">table_ctr</span>, <span class="va">parameters</span><span class="op">$</span><span class="va">out_path</span>, <span class="st">"control"</span><span class="op">)</span></span>
<span><span class="va">exp_input</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF_complexes.html" class="external-link">combine_LR_and_TF_complexes</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">[[</span><span class="st">"PMF_MF2"</span><span class="op">]</span><span class="op">]</span>, <span class="va">table_exp</span>, <span class="va">parameters</span><span class="op">$</span><span class="va">out_path</span>, <span class="st">"PMF_MF2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Second possibility: Directly pass the path to the LR interaction csv files to combine them with the TF interaction results</span></span>
<span><span class="va">ctr_file</span> <span class="op">&lt;-</span> <span class="st">"/path/to/control_lr_results.csv"</span></span>
<span><span class="va">exp_file</span> <span class="op">&lt;-</span> <span class="st">"/path/to/PMF,MF2_lr_results.csv"</span></span>
<span></span>
<span><span class="va">ctr_inptu</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF_complexes.html" class="external-link">combine_LR_and_TF_complexes</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">[[</span><span class="st">"control"</span><span class="op">]</span><span class="op">]</span>, <span class="va">ctr_file</span>, <span class="va">parameters</span><span class="op">$</span><span class="va">out_path</span>, <span class="st">"control"</span><span class="op">)</span></span>
<span><span class="va">exp_input</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF_complexes.html" class="external-link">combine_LR_and_TF_complexes</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">[[</span><span class="st">"PMF_MF2"</span><span class="op">]</span><span class="op">]</span>, <span class="va">exp_file</span>, <span class="va">parameters</span><span class="op">$</span><span class="va">out_path</span>, <span class="st">"PMF_MF2"</span><span class="op">)</span></span></code></pre></div>
<p>Now that there is one input table for each condition of interest,
these can be used to run CrossTalkeR:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"control"</span> <span class="op">=</span> <span class="va">ctr_inptu</span>,</span>
<span>  <span class="st">"PMF_MF2"</span> <span class="op">=</span> <span class="va">exp_input</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">output</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="st">"/path/to/save/folder"</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_report.html">generate_report</a></span><span class="op">(</span><span class="va">paths</span>,</span>
<span>  out_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">output</span>, <span class="st">"/"</span><span class="op">)</span>,</span>
<span>  out_file <span class="op">=</span> <span class="st">"HumanMyfib_TF_example.html"</span>,</span>
<span>  output_fmt <span class="op">=</span> <span class="st">"html_document"</span>,</span>
<span>  report <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  comparison <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PMF_MF2"</span>, <span class="st">"control"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="crosstalker-results">CrossTalkeR Results<a class="anchor" aria-label="anchor" href="#crosstalker-results"></a>
</h3>
<p>Incorporating TFs into the CrossTalkeR analysis introduces greater
complexity to the network, particularly at the CGI level. Since the CCI
component emphasizes intercellular communication, TFs — which contribute
to potential intracellular interactions — are excluded from this aspect
of the analysis, leaving the results unchanged. Therefore, this segment
of the example concentrates on the modifications observed in the CGI
component. Furthermore, we utilize only the results from the comparison
report because condition-specific TFs were used in the CrossTalkeR
analysis.</p>
<p>A first example is to compare the PageRank results of a run with only
LR genes to the current with added TFs. A barplot with the top up and
down PageRank scored genes can be generated like this:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_bar_rankings_cgi</span><span class="op">(</span><span class="va">data</span>, <span class="st">"PMF_MF2_x_control_ggi"</span>, ranking <span class="op">=</span> <span class="st">"Pagerank"</span><span class="op">)</span></span></code></pre></div>
<p>Now we can compare the results from CrossTalkeR with only LR (left)
and the results with TFs (right):</p>
<div style="display: flex; align-items: center; justify-content: center;">
<figure class="figure" style="display: table; margin: auto; height: 80%; width: 80%;"><img src="TF_images%2FPagerank_LR.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><figure class="figure" style="display: table; margin: auto; height: 80%; width: 80%;"><img src="TF_images%2FPagerank_TF.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure>
</div>
<p> </p>
<p>It can be observed that the inclusion of TFs increases the
variability of gene types coming up in the top genes. The ITGB1 receptor
in megakaryocytes remains among the top genes and several TFs are also
present in the results; for example, STAT5B in megakaryocytes and ATF4
in Fibroblasts are among the upregulated genes, while TFAP2A in MSCs is
among the down regulated genes. We can analyze not only the PageRank but
also examine the top genes from other rankings. For this purpose, the
previously used method call can be adjusted:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_bar_rankings_cgi</span><span class="op">(</span><span class="va">data</span>, <span class="st">"PMF_MF2_x_control_ggi"</span>, ranking <span class="op">=</span> <span class="st">"Influencer"</span>, type <span class="op">=</span> <span class="st">"L"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_bar_rankings_cgi</span><span class="op">(</span><span class="va">data</span>, <span class="st">"PMF_MF2_x_control_ggi"</span>, ranking <span class="op">=</span> <span class="st">"Listener"</span>, type <span class="op">=</span> <span class="st">"R"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">plot_bar_rankings_cgi</span><span class="op">(</span><span class="va">data</span>, <span class="st">"PMF_MF2_x_control_ggi"</span>, ranking <span class="op">=</span> <span class="st">"Mediator"</span>, type <span class="op">=</span> <span class="st">"TF"</span><span class="op">)</span></span></code></pre></div>
<p> </p>
<div style="display: flex; align-items: center; justify-content: center;">
<figure class="figure" style="display: table; margin: auto; height: 80%; width: 80%;"><img src="TF_images%2FL_Influencers.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><figure class="figure" style="display: table; margin: auto; height: 80%; width: 80%;"><img src="TF_images%2FR_Listener.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure>
</div>
<figure class="figure" style="display: table; margin: auto; height: 80%; width: 80%;"><img src="TF_images%2FTF_Mediators.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
<p>By integrating intracellular signaling into our analysis, we can now
trace predicted signaling pathways within cells to determine whether
incoming signals induce responses in specific cell types or, conversely,
to identify the upstream regulators of an active ligand. As example, we
can select the top influencer ligand, TGFB1 in megakaryocytes, from the
top listener barplot. We begin by generating a standard LR Sankey plot
to visualize the intercellular interactions involving TGFB1:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">PMF_MF2_x_control</span>,</span>
<span>    target <span class="op">=</span> <span class="st">"TGFB1|L"</span>,</span>
<span>    ligand_cluster <span class="op">=</span> <span class="st">"Megakaryocyte"</span>,</span>
<span>    receptor_cluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    plt_name <span class="op">=</span> <span class="st">"TGFB1 Megakaryocyte Signaling PMF_MF2 vs control"</span>,</span>
<span>    threshold <span class="op">=</span> <span class="fl">50</span>, tfflag <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 90%; width: 90%;"><img src="TF_images%2FTGFB1_signaling_from_Megakaryocytes.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
<p>Now, we are introducing a new graph sankey plot, that shows the
upstream signaling of TGFB1 in megakaryocytes:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">pagerank_table</span> <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">$</span><span class="va">PMF_MF2_x_control_ggi</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"nodes"</span>, <span class="st">"Pagerank"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pagerank_table</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">pagerank_table</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pagerank_table</span><span class="op">)</span> <span class="op">=</span> <span class="va">pagerank_table</span><span class="op">$</span><span class="va">nodes</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_graph_sankey_tf.html">plot_graph_sankey_tf</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">PMF_MF2_x_control</span>,</span>
<span>    <span class="va">pagerank_table</span>,</span>
<span>    target <span class="op">=</span> <span class="st">"TGFB1|L"</span>,</span>
<span>    cluster <span class="op">=</span> <span class="st">"Megakaryocyte"</span>,</span>
<span>    target_type <span class="op">=</span> <span class="st">"Ligand"</span>,</span>
<span>    plt_name <span class="op">=</span> <span class="st">"Signaling Upstream TGFB1 in Megakaryocytes"</span><span class="op">)</span></span></code></pre></div>
<p> </p>
<figure class="figure" style="display: table; margin: auto; height: 90%; width: 90%;"><img src="TF_images%2FTGFB1_upstream_signaling_Megakaryocytes.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
<p>In this plot, we focus on the PageRank of the included genes. Each
node’s weight represents the corresponding PageRank value of each gene,
and the edges are weighted by the PageRank values of the receptors or
ligands involved in the interactions. In the example, multiple TFs
predicted to regulate the TGFB1 gene are shown to be active. Notably,
one of these TFs, MYC, is identified as the top mediator TF. While MYC
was previously described to have a profibrotic role [6], its role in
megakaryocytes has not yet been further studied, suggesting potential
opportunities for investigation.</p>
<p>To follow up on connecting TGFB1 further to upstream signals, we
focus now on the top listener receptor, ITGB1, which is here predicted
to be upstream of MYC in megakaryocytes. Building on this, we examine
the top incoming signals and the cell types from which they originate,
targeting ITGB1 in megakaryocytes:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">PMF_MF2_x_control</span>,</span>
<span>    target <span class="op">=</span> <span class="st">"ITGB1|R"</span>,</span>
<span>    ligand_cluster <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    receptor_cluster <span class="op">=</span> <span class="st">"Megakaryocyte"</span>,</span>
<span>    plt_name <span class="op">=</span> <span class="st">"ITGB1 Megakaryocyte Signaling PMF_MF2 vs control"</span>,</span>
<span>    threshold <span class="op">=</span> <span class="fl">50</span>, tfflag <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 90%; width: 90%;"><img src="TF_images%2FITGB1_receiving_signals_Megakaryocytes.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
<p>If we put together the sepparated signalig plots, we can receive this
signaling chain through the megakaryocyte cells (this image is not
automatically generated):</p>
<figure class="figure" style="display: table; margin: auto; height: 110%; width: 110%;"><img src="TF_images%2FCombined_%20Sankey.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p>[1] Garcia-Alonso L, Holland CH, Ibrahim MM, Turei D, Saez-Rodriguez
J. “Benchmark and integration of resources for the estimation of human
transcription factor activities.” Genome Research. 2019. DOI:
10.1101/gr.240663.118.</p>
<p>[2] A Valdeolivas, D Turei, A Gabor (2019) “OmnipathR: client for the
OmniPath web service.” Bioconductor Package</p>
<p>[3] D Turei, T Korcsmaros and J Saez-Rodriguez (2016) OmniPath:
guidelines and gateway for literature-curated signaling pathway
resources. Nature Methods 13 (12); PMID: 27898060</p>
<p>[4] D Turei, A Valdeolivas, L Gul, N Palacio-Escat, M Klein, O
Ivanova, M Olbei, A Gabor, F Theis, D Modos, T Korcsmaros and J
Saez-Rodriguez (2021) Integrated intra- and intercellular signaling
knowledge for multicellular omics analysis. Molecular Systems Biology
17: e9923; DOI: 10.15252/msb.20209923</p>
<p>[5] James S Nagai, Nils B Leimkühler, Michael T Schaub, Rebekka K
Schneider, Ivan G Costa, CrossTalkeR: analysis and visualization of
ligand–receptorne tworks, Bioinformatics, Volume 37, Issue 22, 15
November 2021, Pages 4263–4265, <a href="https://doi.org/10.1093/bioinformatics/btab370" class="external-link uri">https://doi.org/10.1093/bioinformatics/btab370</a></p>
<p>[6] Zacarías-Fluck MF, Soucek L, Whitfield JR. MYC: there is more to
it than cancer. Front Cell Dev Biol. 2024 Mar 6;12:1342872. DOI:
10.3389/fcell.2024.1342872.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vanessa Kloeker, James Shiniti Nagai, Nils Leimkühler, Michael T. Schaub, Rebekka Schneider, Ivan Costa Gesteira.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
