<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CrossTalkeR">
<title>Transcription Factor Activity Analysis • CrossTalkeR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Transcription Factor Activity Analysis">
<meta property="og:description" content="CrossTalkeR">
<meta property="og:image" content="https://github.com/CostaLab/CrossTalkeR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CrossTalkeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.4.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/CrossTalker_install_basicusage.html">Installation and Usage</a>
    <a class="dropdown-item" href="../articles/HumanFibrosis.html">Human Myelofibrosis example</a>
    <a class="dropdown-item" href="../articles/LR2TF_analysis.html">Transcription Factor Analysis</a>
    <a class="dropdown-item" href="../articles/run_liana.html">liana Tutorial</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="https://github.com/CostaLab/CrossTalkeR">
    <span class="fab fa fab fa fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="LR2TF_analysis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Transcription Factor Activity Analysis</h1>
                        <h4 data-toc-skip class="author">Vanessa Klöker</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><h4 data-toc-skip class="author">James S. Nagai</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><h4 data-toc-skip class="author">Ivan G. Costa</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH Aachen University, Aachen, 52074 Germany<br><small class="dont-index">Source: <a href="https://github.com/CostaLab/CrossTalkeR/blob/HEAD/vignettes/LR2TF_analysis.rmd"><code>vignettes/LR2TF_analysis.rmd</code></a></small>
      <div class="d-none name"><code>LR2TF_analysis.rmd</code></div>
    </address>
</address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="teanscription-factor-activity-analysis-on-human-myelofibrosis-scrna-seq-data-using-the-lr2tf-package-and-crosstalker">Teanscription Factor Activity Analysis on Human Myelofibrosis scRNA-seq Data Using the LR2TF Package and CrossTalkeR<a class="anchor" aria-label="anchor" href="#teanscription-factor-activity-analysis-on-human-myelofibrosis-scrna-seq-data-using-the-lr2tf-package-and-crosstalker"></a>
</h2>
<p>This is a guide for the usage of the LR2TF package to infer TF activities from scRNA-seq data. The transcription factor activities are estimated using the decoupleR tool. Using the DoRothEA regulon version from decoupleR and post-translational interactions from the Omnipath database[2,3,4], connections are made between transcription factors and ligands and receptors.</p>
<p>The results can be combined with ligand-receptor interactions and then analyzed using CrossTalkeR (<a href="https://github.com/CostaLab/CrossTalkeR/" class="uri">https://github.com/CostaLab/CrossTalkeR/</a>) [5].</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>The package can be installed directly from github:  </p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"CostaLab/LR2TF"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"CostaLab/LR2TF"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p> </p>
</div>
<div class="section level3">
<h3 id="transcription-factor-activity-predictions-using-decoupler-in-python">Transcription factor activity predictions using decoupleR in python<a class="anchor" aria-label="anchor" href="#transcription-factor-activity-predictions-using-decoupler-in-python"></a>
</h3>
<p>For the transcription factor activity predictions we recommend using the decoupleR package in its python version. This provides a better performance, especially for larger data sets. To perform the predictions with decoupleR, we need an AnnData (.h5ad) file. If the scRNA-seq data is saved as a Seurat object (.Rds file), it can be converted to AnnData format using the sceasy package and the function we provide:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">LR2TF</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">#load own dataset:</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"/path/to/seurat_object.Rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#test dataset from package:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">bone_marrow_stromal_cell_example</span>, package <span class="op">=</span> <span class="st">"LR2TF"</span><span class="op">)</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="va">bone_marrow_stromal_cell_example</span></span>
<span></span>
<span><span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/convert_seurat_to_anndata.html" class="external-link">convert_seurat_to_anndata</a></span><span class="op">(</span><span class="va">seurat_object</span>, <span class="st">"/path/to/save/AnnData/object/"</span><span class="op">)</span></span></code></pre></div>
<p>The file “anndata_object.h5ad” will be saved into the user defined path and then used to perform the predictions. Beside the scRNA-seq data file, we also need to define a regulon database in form of a csv file with the coloumns “source”, “target” and “weight”. Within this package we provide the dorothea databases for human and mouse, downloaded from the decoupleR package. These files also contain the column “confidence” (levels A to D) with information on how well described a transcription factor and target gene interaction is in different resources. We recommend using the confidence levels A and B.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">regulon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"regulons"</span>, <span class="st">"human_dorothea_reg.csv"</span>, package <span class="op">=</span> <span class="st">"LR2TF"</span><span class="op">)</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">filtered_regulon</span> <span class="op">&lt;-</span> <span class="va">regulon</span><span class="op">[</span><span class="va">regulon</span><span class="op">$</span><span class="va">confidence</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"B"</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="st">"/path/to/save/filterd_regulon.csv"</span><span class="op">)</span></span></code></pre></div>
<p>Now it is possible to run the transcription factor activity prediction with decoupleR:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true"></a><span class="im">import</span> decoupler <span class="im">as</span> dc</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true"></a>ann_data <span class="op">=</span> sc.read_h5ad(<span class="st">"/path/to/AnnData/object/anndata_object.h5ad"</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true"></a>reg <span class="op">=</span> pd.read_csv(<span class="st">"/path/to/filterd_regulon.csv"</span>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true"></a>dc.run_viper(mat<span class="op">=</span>ann_data, net<span class="op">=</span>reg, source<span class="op">=</span><span class="st">'source'</span>, target<span class="op">=</span><span class="st">'target'</span>, weight<span class="op">=</span><span class="st">'weight'</span>, verbose<span class="op">=</span><span class="va">True</span>, use_raw<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true"></a>estimates <span class="op">=</span> ann_data.obsm[<span class="st">'viper_estimate'</span>]</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true"></a>estimates.to_csv(<span class="st">"/path/to/save/decoupler_results.csv"</span>)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="using-the-lr2tf-package">Using the LR2TF package<a class="anchor" aria-label="anchor" href="#using-the-lr2tf-package"></a>
</h3>
<p>Now, that we have a transcription factor activity matrix, we can continue the analysis with the LR2TF package. In this case we will use our test data for this example. First of all, it is necessary to define the following parameters in form of a list:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parameters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"out_path"</span> <span class="op">=</span> <span class="st">"/path/to/save/results"</span>,</span>
<span>                   reg <span class="op">=</span> <span class="st">"/path/to/filterd_regulon.csv"</span>,</span>
<span>                   <span class="st">"organism"</span> <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>                   <span class="st">"celltype"</span> <span class="op">=</span> <span class="st">"new_annotation"</span>, <span class="co">#name of the meta data field defining cell identities</span></span>
<span>                   <span class="st">"condition"</span> <span class="op">=</span> <span class="st">"protocol"</span>, <span class="co">#name of the meta data field defining conditions</span></span>
<span>                   <span class="st">"comparison_list"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PMF,MF2"</span>, <span class="st">"control"</span><span class="op">)</span><span class="op">)</span>, <span class="co">#list of condition comparison to consider</span></span>
<span>                   <span class="st">"logfc"</span> <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                   <span class="st">"pval"</span> <span class="op">=</span> <span class="fl">0.05</span>, <span class="co">#thresholds for logfc and pval used in differential transcription factor analysis</span></span>
<span>                   <span class="st">"num_cell_filter"</span> <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="co">#define the minimum number of cells per cluster to perform the analysis on</span></span></code></pre></div>
<p>After defining the necessary parameter the transcription factor activity can be performed by calling:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/tf_activity_analysis.html" class="external-link">tf_activity_analysis</a></span><span class="op">(</span>seuratobject <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>                                       tf_activities <span class="op">=</span> <span class="st">"/path/to/decoupler_results.csv"</span>,</span>
<span>                                       arguments_list <span class="op">=</span> <span class="va">parameters</span><span class="op">)</span></span></code></pre></div>
<p>The “results” object contains the results of the performed analyses, consisting of multiple tables inside the object:</p>
<ol style="list-style-type: decimal">
<li>
<strong>tf_activities_condition:</strong> Tables with condition significant transcription factors for each compared condition</li>
<li>
<strong>tf_activities_cluster:</strong> Tables with cluster specific transcription factors for all conditions in the data</li>
<li>
<strong>average_gene_expression:</strong> Matrices for each condition with average gene expressions</li>
<li>
<strong>regulon:</strong> Regulon used for the analysis as specified by the user</li>
<li>
<strong>CTR_input_condition:</strong> for each condition a table with receptor-transcription factor and transcription factor-ligand interactions based on condition specific transcription factors (input for CrossTalker)</li>
<li>
<strong>CTR_input_cluster:</strong> for each condition a table with receptor-transcription factor and transcription factor-ligand interactions based on cluster specific transcription factors (input for CrossTalker)</li>
<li>
<strong>intracellular_network_condition:</strong> for each condition a table with receptor-transcription factor and transcription factor-target gene interactions based on condition specific transcription factors</li>
<li>
<strong>intracellular_network_cluster:</strong> for each condition a table with receptor-transcription factor and transcription factor-target gene interactions based on cluster specific transcription factors</li>
</ol>
<p>(Note that special characters might be exchanged by underscores, if they cause problems with the naming of the tables; eg PMF,MF2 -&gt; PMF_MF2)</p>
<p>The last step is to combine previous results from ligand receptor interaction analyses (e.g. CellPhoneDB) with the transcription factor results. (In the case of the test data the ligand-receptor interactions are provided within the CrossTalkeR package.)</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#First possibility: Read the csv files with LR interaction results and then combine them with the TF interaction results</span></span>
<span><span class="va">table_ctr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"/path/to/control_lr_results.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">table_exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"/path/to/PMF,MF2_lr_results.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ctr_inptu</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF.html" class="external-link">combine_LR_and_TF</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">[[</span><span class="st">"control"</span><span class="op">]</span><span class="op">]</span>, <span class="va">table_ctr</span>, <span class="va">parameters</span><span class="op">$</span><span class="va">out_path</span>, <span class="st">"control"</span><span class="op">)</span></span>
<span><span class="va">exp_input</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF.html" class="external-link">combine_LR_and_TF</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">[[</span><span class="st">"PMF_MF2"</span><span class="op">]</span><span class="op">]</span>, <span class="va">table_exp</span>, <span class="va">parameters</span><span class="op">$</span><span class="va">out_path</span>, <span class="st">"PMF_MF2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Second possibility: Directly pass the path to the LR interaction csv files to combine them with the TF interaction results</span></span>
<span><span class="va">ctr_file</span> <span class="op">&lt;-</span> <span class="st">"/path/to/control_lr_results.csv"</span></span>
<span><span class="va">exp_file</span> <span class="op">&lt;-</span> <span class="st">"/path/to/PMF,MF2_lr_results.csv"</span></span>
<span></span>
<span><span class="va">ctr_inptu</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF.html" class="external-link">combine_LR_and_TF</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">[[</span><span class="st">"control"</span><span class="op">]</span><span class="op">]</span>, <span class="va">ctr_file</span>, <span class="va">parameters</span><span class="op">$</span><span class="va">out_path</span>, <span class="st">"control"</span><span class="op">)</span></span>
<span><span class="va">exp_input</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF.html" class="external-link">combine_LR_and_TF</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">[[</span><span class="st">"PMF_MF2"</span><span class="op">]</span><span class="op">]</span>, <span class="va">exp_file</span>, <span class="va">parameters</span><span class="op">$</span><span class="va">out_path</span>, <span class="st">"PMF_MF2"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-crosstalker">Run CrossTalkeR<a class="anchor" aria-label="anchor" href="#run-crosstalker"></a>
</h3>
<p>Now that there is one input table for each condition of interest, these can be used to run CrossTalkeR:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR">CrossTalkeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'control'</span> <span class="op">=</span> <span class="va">ctr_inptu</span>,</span>
<span>           <span class="st">'PMF_MF2'</span> <span class="op">=</span> <span class="va">exp_input</span><span class="op">)</span></span>
<span></span>
<span><span class="va">output</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="st">"/path/to/save/folder"</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_report.html">generate_report</a></span><span class="op">(</span><span class="va">paths</span>,</span>
<span>                out_path<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">output</span>,<span class="st">'/'</span><span class="op">)</span>,</span>
<span>                out_file <span class="op">=</span> <span class="st">'HumanMyfib_TF_example.html'</span>,</span>
<span>                output_fmt <span class="op">=</span> <span class="st">"html_document"</span>,</span>
<span>                report <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                comparison <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'PMF_MF2'</span>, <span class="st">'control'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p>[1] Garcia-Alonso L, Holland CH, Ibrahim MM, Turei D, Saez-Rodriguez J. “Benchmark and integration of resources for the estimation of human transcription factor activities.” Genome Research. 2019. DOI: 10.1101/gr.240663.118.</p>
<p>[2] A Valdeolivas, D Turei, A Gabor (2019) “OmnipathR: client for the OmniPath web service.” Bioconductor Package</p>
<p>[3] D Turei, T Korcsmaros and J Saez-Rodriguez (2016) OmniPath: guidelines and gateway for literature-curated signaling pathway resources. Nature Methods 13 (12); PMID: 27898060</p>
<p>[4] D Turei, A Valdeolivas, L Gul, N Palacio-Escat, M Klein, O Ivanova, M Olbei, A Gabor, F Theis, D Modos, T Korcsmaros and J Saez-Rodriguez (2021) Integrated intra- and intercellular signaling knowledge for multicellular omics analysis. Molecular Systems Biology 17: e9923; DOI: 10.15252/msb.20209923</p>
<p>[5] James S Nagai, Nils B Leimkühler, Michael T Schaub, Rebekka K Schneider, Ivan G Costa, CrossTalkeR: analysis and visualization of ligand–receptorne tworks, Bioinformatics, Volume 37, Issue 22, 15 November 2021, Pages 4263–4265, <a href="https://doi.org/10.1093/bioinformatics/btab370" class="external-link uri">https://doi.org/10.1093/bioinformatics/btab370</a></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vanessa Kloeker, James Shiniti Nagai, Nils Leimkühler, Michael T. Schaub, Rebekka Schneider, Ivan Costa Gesteira.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
