<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>CrossTalkeR Example - Human Myocardial Infarction • CrossTalkeR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="CrossTalkeR Example - Human Myocardial Infarction">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-black" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CrossTalkeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/CrossTalkeR.html">Get started</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="../articles/CrossTalker_basicusage.html">CrossTalkeR 101</a></li>
    <li><a class="dropdown-item" href="../articles/HumanFibrosis.html">Example: Bone Marrow Fibrosis in Human (Leimkuhler et. al., 2021)</a></li>
    <li><a class="dropdown-item" href="../articles/MyocardialInfarction.html">Example: Heart Myocardial Infarcition in Human (Kuppe et. al., 2022)</a></li>
    <li><a class="dropdown-item" href="../articles/LR2TF_analysis.html">Intracellular Communication Analysis Bone Marrow Fibrosis in Human</a></li>
    <li><a class="dropdown-item" href="../articles/run_liana.html">Ligand-Receptor Inference with Liana</a></li>
    <li><a class="dropdown-item" href="../articles/ProgenyLRExample.html">Ligand-Receptor Pathway Enrichment with Progeny - Bone Marrow Fibrosis in Human</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/Install.html">Install</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CostaLab/CrossTalkeR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>CrossTalkeR Example - Human Myocardial Infarction</h1>
                        <h4 data-toc-skip class="author">Vanessa
Klöker</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074
Germany<br><h4 data-toc-skip class="author">James
S. Nagai</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074
Germany<br><h4 data-toc-skip class="author">Ivan
G. Costa</h4>
            <address class="author_afil">
      Institute for Computational Genomics, Faculty of Medicine, RWTH
Aachen University, Aachen, 52074 Germany<br><small class="dont-index">Source: <a href="https://github.com/CostaLab/CrossTalkeR/blob/HEAD/vignettes/MyocardialInfarction.rmd" class="external-link"><code>vignettes/MyocardialInfarction.rmd</code></a></small>
      <div class="d-none name"><code>MyocardialInfarction.rmd</code></div>
    </address>
</address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="cell-cell-communication-analysis-of-human-myelofibrosis-scrna-seq-data-using-crosstalker">Cell-Cell Communication Analysis of Human Myelofibrosis scRNA-seq
Data Using CrossTalkeR<a class="anchor" aria-label="anchor" href="#cell-cell-communication-analysis-of-human-myelofibrosis-scrna-seq-data-using-crosstalker"></a>
</h2>
<p>Here, we demonstrate the usage of CrossTalkeR on the case study of
human myocardian infarction snRNA-seq data from the Nature paper
“Spatial multi-omic map of human myocardial infarction” (Kuppe et al.).
Aim of the study was to build a high-resolution map of human cardiac
remodelling after myocardial infarction to provide a better
understanding of these remodelling processes. We are focusing here on
the conditions “myogenic” (control condition) and “ischemic” (disease
condition) and go through the CrossTalkeR results step by step. The
original paper already contains CrossTalkeR results and in a later part
of this tutorial we will replicate a few of the results.</p>
<div class="section level3">
<h3 id="ligand-receptor-interaction-prediction-with-liana">Ligand-Receptor Interaction Prediction with LIANA<a class="anchor" aria-label="anchor" href="#ligand-receptor-interaction-prediction-with-liana"></a>
</h3>
<p>Before we can run CrossTalkeR, we need the prediction of
ligand-receptor interaction for our data set. Due to the large sample
size we are using the python version of LIANA for a better
performance:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="im">import</span> liana <span class="im">as</span> li</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>out_path <span class="op">=</span> <span class="st">"/out/path/"</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>data <span class="op">=</span> sc.read_h5ad(<span class="st">"heart_data_object.h5ad"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>data.raw <span class="op">=</span> data</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="bu">set</span>(data.obs[<span class="st">'sampleType'</span>]):</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>    <span class="bu">print</span>(i)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>    lr<span class="op">=</span>li.method.cellphonedb(data[data.obs[<span class="st">'sampleType'</span>]<span class="op">==</span>i],</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>                        groupby<span class="op">=</span><span class="st">'cell_subtype2'</span>,</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>                        expr_prop<span class="op">=</span><span class="fl">0.1</span>,</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>                        verbose<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>                        resource_name<span class="op">=</span><span class="st">'consensus'</span>,inplace<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a>    lr.to_csv(<span class="ss">f"</span><span class="sc">{</span>out_path<span class="sc">}{</span>i<span class="sc">}</span><span class="ss">_lr_liana_consensus.csv"</span>)</span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> os.listdir(out_path):</span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a>    <span class="cf">if</span> i.endswith(<span class="st">'lr_liana_consensus.csv'</span>):</span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a>        evfull <span class="op">=</span> pd.read_csv(out_path<span class="op">+</span>i)</span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a>        evfull <span class="op">=</span> evfull.loc[:,[<span class="st">'ligand'</span>,<span class="st">'receptor_complex'</span>,<span class="st">'source'</span>,<span class="st">'target'</span>,<span class="st">'lr_means'</span>,<span class="st">'cellphone_pvals'</span>]]</span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a>        evfull[<span class="st">'type_gene_A'</span>] <span class="op">=</span> <span class="st">'Ligand'</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a>        evfull[<span class="st">'type_gene_B'</span>] <span class="op">=</span> <span class="st">'Receptor'</span></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a>        evfull[<span class="st">'gene_A'</span>] <span class="op">=</span> evfull[<span class="st">'ligand'</span>]</span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a>        evfull[<span class="st">'gene_B'</span>] <span class="op">=</span> evfull[<span class="st">'receptor_complex'</span>]</span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a>        evfull[<span class="st">'MeanLR'</span>] <span class="op">=</span> evfull[<span class="st">'lr_means'</span>]</span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a>        k <span class="op">=</span> i[<span class="dv">0</span>:i.find(<span class="st">'_lr_'</span>)]</span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a>        evfull <span class="op">=</span> evfull.loc[<span class="bu">list</span>(evfull.cellphone_pvals.to_numpy()<span class="op">&lt;=</span><span class="fl">0.05</span>),:]</span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a>        evfull <span class="op">=</span> evfull.loc[:, [<span class="st">'source'</span>, <span class="st">'target'</span>, <span class="st">'type_gene_A'</span>, <span class="st">'type_gene_B'</span>, <span class="st">'gene_A'</span>, <span class="st">'gene_B'</span>, <span class="st">'MeanLR'</span>]].to_csv(<span class="ss">f'</span><span class="sc">{</span>out_path<span class="sc">}{</span>k<span class="sc">}</span><span class="ss">_lr_ready.csv'</span>)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="intracellular-communication-analysis-with-intratalker">Intracellular Communication Analysis with IntraTalker<a class="anchor" aria-label="anchor" href="#intracellular-communication-analysis-with-intratalker"></a>
</h3>
<p>We can not only analyze the ligand-receptor interactions but also the
intracellular communication. For this we use transcription factor
activity predictions from decoupler and the IntraTalker package. Similar
to usage of LIANA we are using the python version of decoupler:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="im">import</span> decoupler <span class="im">as</span> dc</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="bu">file</span> <span class="op">=</span> <span class="st">"heart_data_object.h5ad"</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>reg <span class="op">=</span> pd.read_csv(<span class="st">"regulon_dorothea_human_AB.csv"</span>)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>ann_data <span class="op">=</span> sc.read_h5ad(<span class="bu">file</span>)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>dc.run_ulm(</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>    mat<span class="op">=</span>ann_data,</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>    net<span class="op">=</span>reg,</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>    source<span class="op">=</span><span class="st">'source'</span>,</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>    target<span class="op">=</span><span class="st">'target'</span>,</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>    weight<span class="op">=</span><span class="st">'weight'</span>,</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>    verbose<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>    use_raw<span class="op">=</span><span class="va">False</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>)</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a>estimates <span class="op">=</span>ann_data.obsm[<span class="st">'ulm_estimate'</span>]</span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a>estimates.to_csv(<span class="st">"/out/path/decoupler_dorotheaAB_results_ulm.csv"</span>)</span></code></pre></div>
<p>To find the transcription factors that show a significant different
between the conditions and to connect the transcription factors to
upstream receptors and downstream ligands we use the IntraTalker
package. First we generate the intracellular communication tables
based:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.igraph.org/" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">LR2TF</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/CostaLab/CrossTalkeR" class="external-link">CrossTalkeR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">heart_data_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"heart_data_object.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">parameters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"out_path"</span> <span class="op">=</span> <span class="st">"/out/path/"</span>,</span>
<span>                   reg <span class="op">=</span> <span class="st">"regulon_dorothea_human_AB.csv"</span>,</span>
<span>                   <span class="st">"organism"</span> <span class="op">=</span> <span class="st">"human"</span>,</span>
<span>                   <span class="st">"celltype"</span> <span class="op">=</span> <span class="st">"cell_subtype2"</span>,</span>
<span>                   <span class="st">"condition"</span> <span class="op">=</span> <span class="st">"sampleType"</span>,</span>
<span>                   <span class="st">"comparison_list"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ischemic"</span>, <span class="st">"myogenic"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                   <span class="st">"logfc"</span> <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                   <span class="st">"pval"</span> <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu">tf_activity_analysis</span><span class="op">(</span>seuratobject <span class="op">=</span> <span class="va">heart_data_object</span>,</span>
<span>                                       tf_activities <span class="op">=</span> <span class="st">"/out/path/decoupler_dorotheaAB_results_ulm.csv"</span>,</span>
<span>                                       arguments_list <span class="op">=</span> <span class="va">parameters</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="input-preparation-for-crosstalker">Input Preparation for CrossTalkeR<a class="anchor" aria-label="anchor" href="#input-preparation-for-crosstalker"></a>
</h3>
<p>In the next step, we combine the ligand-receptor and intracellular
communication predictions to generate the input for CrossTalkeR:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ischemic_LR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"/out/path/ischemic_lr_ready.csv"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">myogenic_LR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"/out/path/myogenic_lr_ready.csv"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ischemic_combined</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF_complexes.html" class="external-link">combine_LR_and_TF_complexes</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">$</span><span class="va">ischemic</span>,</span>
<span>                                                     <span class="va">ischemic_LR</span>,</span>
<span>                                                     <span class="st">"/out/path/"</span>,</span>
<span>                                                     <span class="st">"ischemic"</span><span class="op">)</span></span>
<span><span class="va">myogenic_combined</span> <span class="op">&lt;-</span> <span class="fu">LR2TF</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/LR2TF/man/combine_LR_and_TF_complexes.html" class="external-link">combine_LR_and_TF_complexes</a></span><span class="op">(</span><span class="va">results</span><span class="op">@</span><span class="va">CTR_input_condition</span><span class="op">$</span><span class="va">myogenic</span>,</span>
<span>                                                     <span class="va">myogenic_LR</span>,</span>
<span>                                                     <span class="st">"/out/path/"</span>,</span>
<span>                                                     <span class="st">"myogenic"</span><span class="op">)</span></span></code></pre></div>
<p>now the input data for CrossTalkeR is ready and we can run the
analysis.</p>
</div>
<div class="section level3">
<h3 id="running-crosstalker">Running CrossTalkeR<a class="anchor" aria-label="anchor" href="#running-crosstalker"></a>
</h3>
<p>The last step is to run CrossTalkeR:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'myogenic'</span> <span class="op">=</span> <span class="va">myogenic_combined</span>,              </span>
<span>              <span class="st">'ischemic'</span> <span class="op">=</span> <span class="va">ischemic_combined</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_report.html">generate_report</a></span><span class="op">(</span><span class="va">paths</span>,</span>
<span>                            out_path <span class="op">=</span> <span class="st">"/out/path/"</span>,</span>
<span>                            threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                            out_file <span class="op">=</span> <span class="st">'myocardial_infarction.html'</span>,</span>
<span>                            output_fmt <span class="op">=</span> <span class="st">"html_document"</span>,</span>
<span>                            org <span class="op">=</span> <span class="st">"hsa"</span>,</span>
<span>                            comparison <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ischemic'</span>, <span class="st">'myogenic'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                            filtered_net <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p> </p>
</div>
</div>
<div class="section level2">
<h2 id="crosstalker-results-ischemic-vs-myogenic-condition">CrossTalkeR Results Ischemic vs Myogenic condition<a class="anchor" aria-label="anchor" href="#crosstalker-results-ischemic-vs-myogenic-condition"></a>
</h2>
<p>The second part of this tutorial focuses on the results of the
comparative analysis of ischemic versus myogenic conditions.</p>
<div class="section level3">
<h3 id="cci-results">CCI Results<a class="anchor" aria-label="anchor" href="#cci-results"></a>
</h3>
<p>We start our result analysis with the cell-cell interaction (CCI)
analysis of CrossTalkeR and therefore take a look at our CCI network
plot:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_cci.html">plot_cci</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">ischemic_x_myogenic</span>,</span>
<span>        colors <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">colors</span>,</span>
<span>        plt_name <span class="op">=</span> <span class="st">"ischemic vs myogenic"</span>,</span>
<span>        coords <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">coords</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">ischemic_x_myogenic</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,<span class="op">]</span>,</span>
<span>        emax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        leg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        high <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        ignore_alpha <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        log <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        efactor <span class="op">=</span> <span class="fl">8</span>,</span>
<span>        vfactor <span class="op">=</span> <span class="fl">12</span>, </span>
<span>        pg <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">[[</span><span class="st">"ischemic_x_myogenic"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">Pagerank</span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="heart_images%2FCCI_heart_unfiltered.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p>In ischemic conditions, CCIs are generally reduced. However, certain
cell types remain notably important, as highlighted by their higher
Pagerank scores. These key cell types include cardiomyocyte clusters,
two of the four fibroblast clusters, and specific myeloid cells, such as
SPP1-expressing macrophages and mast cells. These findings indicate that
despite the reduced overall interactions, these cell types are predicted
to have a central role in the CCI network of the ischemic condition.</p>
<p>CrossTalkeR also offers a refined CCI network by employing Fisher’s
exact test to filter the network based on interaction proportions. We
can plot the resulting statistics in a Violon Plot:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">EnhancedVolcano</span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">stats</span><span class="op">$</span><span class="va">ischemic_x_myogenic</span>,</span>
<span>                lab <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">stats</span><span class="op">$</span><span class="va">ischemic_x_myogenic</span><span class="op">$</span><span class="va">columns_name</span>,</span>
<span>                x <span class="op">=</span> <span class="st">"lodds"</span>,</span>
<span>                y <span class="op">=</span> <span class="st">"p"</span>,</span>
<span>                pCutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="heart_images%2FCCI_Volcano.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p>Statistically significant points are labeled with the names of the
corresponding edge pairs, with the significance threshold set to a
p-value greater than 0.05 and a log2 fold change above one. We see that
many of these significant interactions involve high Pagerank cell types,
including fibroblast, cardiomyocyte, and myeloid cell clusters. Using
this threshold (the default in CrossTalkeR), we can retain only
interacting cell pairs with a p-value greater than 0.05 as an example,
and replot the CCI network:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_cci.html">plot_cci</a></span><span class="op">(</span>graph <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">ischemic_x_myogenic_filtered</span>,</span>
<span>        colors <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">colors</span>,</span>
<span>        plt_name <span class="op">=</span> <span class="st">"ischemic vs myogenic filtered"</span>,</span>
<span>        coords <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">coords</span><span class="op">[</span><span class="fu"><a href="https://r.igraph.org/reference/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">graphs</span><span class="op">$</span><span class="va">ischemic_x_myogenic_filtered</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,<span class="op">]</span>,</span>
<span>        emax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>        leg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        high <span class="op">=</span> <span class="fl">0</span>,</span>
<span>        ignore_alpha <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        log <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        efactor <span class="op">=</span> <span class="fl">8</span>,</span>
<span>        vfactor <span class="op">=</span> <span class="fl">12</span>,</span>
<span>        pg <span class="op">=</span> <span class="va">data</span><span class="op">@</span><span class="va">rankings</span><span class="op">[[</span><span class="st">"ischemic_x_myogenic_filtered"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">Pagerank</span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="heart_images%2FCCI_heart_filtered.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p>In the filtered CCI plot, positive edges become more visible, though
all cell types remain except for the Neuronal and Purkinje_fibers
clusters. The Pagerank of the nodes remains largely unchanged.</p>
<p>To confirm the top cell types identified by Pagerank, we can examine
their Pagerank scores in more detail using a bar plot:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"ischemic_x_myogenic_filtered"</span>, <span class="st">"Pagerank"</span>, type <span class="op">=</span> <span class="cn">NULL</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="heart_images%2FCCI_barplot_Pagerank.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p>We can further take a look at the Influencer and Listener ranking to
observe which cell types are more sending or more receiving:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"ischemic_x_myogenic_filtered"</span>, <span class="st">"Influencer"</span>, type <span class="op">=</span> <span class="cn">NULL</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"ischemic_x_myogenic_filtered"</span>, <span class="st">"Listener"</span>, type <span class="op">=</span> <span class="cn">NULL</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<div style="display: flex; align-items: center; justify-content: center;">
<figure class="figure" style="display: table; margin: auto; height: 60%; width: 60%;"><img src="heart_images%2FCCI_barplot_Influencer.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><figure class="figure" style="display: table; margin: auto; height: 60%; width: 60%;"><img src="heart_images%2FCCI_barplot_Listener.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure>
</div>
<p>These plots highlight that nearly all cell types are predicted to
communicate less in the ischemic condition compared to the myogenic
condition. Exceptions include the Prolif, Lymphatic_Endo, Fib2_Myofib,
and Fib3_C7 clusters, which exhibit a higher Influencer Score in the
ischemic condition (indicating more outgoing signals). Additionally, the
Lymphatic_Endo, damaged_CM, and Mast clusters show a higher Listener
Score, indicating more incoming signals in ischemia.</p>
<p>Most of the observations at the CCI level are consistent with the
findings in the paper. For instance, fibrotic structures are more
prominent in the ischemic condition, which is also reflected in the
Pagerank scores, as at least two out of the four fibroblast clusters
have a higher Pagerank in the ischemic condition.</p>
</div>
<div class="section level3">
<h3 id="cgi-results">CGI Results<a class="anchor" aria-label="anchor" href="#cgi-results"></a>
</h3>
<p>Next, we shift to the gene-cell interaction (GCI) level. Similar to
the CCI level, we can examine the network’s topological rankings. We
begin by reviewing the top Pagerank results, which identify genes within
a specific cell type that are important in the network’s signaling. For
now, we will focus exclusively on ligand-receptor (LR) interactions.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"ischemic_x_myogenic_ggi"</span>, <span class="st">"Pagerank"</span>, type <span class="op">=</span> <span class="st">"LR"</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span>, mode <span class="op">=</span> <span class="st">"cgi"</span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="heart_images%2FLR_barplot_Pagerank.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
<p>In the top genes, we find both ligands (L) and receptors (R). Many of
these genes are involved in immune system processes, such as integrin
receptors (e.g., ITGA10_ITGB1, ITGA9, ITGB1, and others). These are
predicted to play a larger role in certain clusters (e.g., Venous_Endo,
Lymphatic_Endo, CD_8) or a smaller role in others (e.g., Pericyte_1,
CD_4, Pericyte_2). Additionally, the TGFB1 ligand is among the top genes
in the Fib3_C7 cluster, and the LRP1 receptor is found in
SPP1_Macrophages.</p>
<p>To gain a clearer understanding of outgoing and incoming signals, we
examine the ligands with the top Influencer scores and the receptors
with the top Listener scores:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"ischemic_x_myogenic_ggi"</span>, <span class="st">"Influencer"</span>, type <span class="op">=</span> <span class="st">"L"</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span>, mode <span class="op">=</span> <span class="st">"cgi"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_bar_rankings.html">plot_bar_rankings</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"ischemic_x_myogenic_ggi"</span>, <span class="st">"Listener"</span>, type <span class="op">=</span> <span class="st">"R"</span>, filter_sign <span class="op">=</span> <span class="cn">NULL</span>, mode <span class="op">=</span> <span class="st">"cgi"</span><span class="op">)</span></span></code></pre></div>
<div style="display: flex; align-items: center; justify-content: center;">
<figure class="figure" style="display: table; margin: auto; height: 60%; width: 60%;"><img src="heart_images%2FL_barplot_Influencer.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><figure class="figure" style="display: table; margin: auto; height: 60%; width: 60%;"><img src="heart_images%2FR_barplot_Listener.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure>
</div>
<p> </p>
<p>The list of top influencers mainly includes genes important in wound
healing, fibrosis, and immune response (e.g., TGFB1, collagens, etc.).
The receptors are primarily integrin receptors and subunits and CD44,
which play key roles in immune response and tissue repair processes.
These results, which have already been described in the paper, further
support these findings.</p>
<p>Next, we create the Sankey plots for the ischemic vs. myogenic
comparisons, as shown in extended figures 12d and e in the original
paper. It is important to note that these plots may differ slightly, as
we are using newer versions of the LIANA package and, for simplicity, we
are considering fewer samples in this tutorial.</p>
<p>We begin with the Sankey plots showing signaling between the Fib2 and
SPP1_Macrophages clusters (in both directions):</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">ischemic_x_myogenic</span>,</span>
<span>    target <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    ligand_cluster <span class="op">=</span> <span class="st">"Fib2_Myofib"</span>,</span>
<span>    receptor_cluster <span class="op">=</span> <span class="st">"SPP1_Macrophages"</span>,</span>
<span>    plt_name <span class="op">=</span> <span class="st">"Fib2_Myofib to SPP1_Macrophages Signaling ischemic vs myogenic"</span>,</span>
<span>    threshold <span class="op">=</span> <span class="fl">50</span>, </span>
<span>    tfflag <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="heart_images%2FFib2_SPP1_Mac_Sankey.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">ischemic_x_myogenic</span>,</span>
<span>    target <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    ligand_cluster <span class="op">=</span> <span class="st">"SPP1_Macrophages"</span>,</span>
<span>    receptor_cluster <span class="op">=</span> <span class="st">"Fib2_Myofib"</span>,</span>
<span>    plt_name <span class="op">=</span> <span class="st">"SPP1_Macrophages to Fib2_Myofib Signaling ischemic vs myogenic"</span>,</span>
<span>    threshold <span class="op">=</span> <span class="fl">50</span>, </span>
<span>    tfflag <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="heart_images%2FSPP1_Mac_Fib2_Sankey.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
<p>Additionally, a Sankey plot was created for a subset of cell types,
displaying only interactions involving the TGFB1 ligand:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"LYVE_FOLR_Macrophages"</span>,</span>
<span><span class="st">"LYVE_PLTP_Macrophages"</span>,</span>
<span><span class="st">"CCL18_Macrophages"</span>,</span>
<span><span class="st">"SPP1_Macrophages"</span>,</span>
<span><span class="st">"Monocytes"</span>,</span>
<span><span class="st">"Fib4_COL15A1"</span>,</span>
<span><span class="st">"Fib3_C7"</span>,</span>
<span><span class="st">"Fib2_Myofib"</span>,</span>
<span><span class="st">"Fib1_SCARA5"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_sankey.html">plot_sankey</a></span><span class="op">(</span><span class="va">data</span><span class="op">@</span><span class="va">tables</span><span class="op">$</span><span class="va">ischemic_x_myogenic</span>,</span>
<span>    target <span class="op">=</span> <span class="st">"TGFB1|L"</span>,</span>
<span>    ligand_cluster <span class="op">=</span> <span class="va">subset</span>,</span>
<span>    receptor_cluster <span class="op">=</span> <span class="va">subset</span>,</span>
<span>    plt_name <span class="op">=</span> <span class="st">"TGFB1 Signaling Between Cell Type Subset ischemic vs myogenic"</span>,</span>
<span>    threshold <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    tfflag <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<figure class="figure" style="display: table; margin: auto; height: 100%; width: 100%;"><img src="heart_images%2Fsubset_TGFB1_Sankey.png" alt=""><figcaption class="figcaption" style="display: table-caption; caption-side: bottom;"></figcaption></figure><p> </p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Vanessa Kloeker, James Shiniti Nagai, Nils Leimkühler, Michael T. Schaub, Rebekka Schneider, Ivan Costa Gesteira.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
