<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run all LR Downstream analysis — generate_report • CrossTalkeR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run all LR Downstream analysis — generate_report" />
<meta property="og:description" content="This function loads the single conditions LR outputs and return the LR network based analysis.
It assumes that the table present the following columns Ligand, Ligand.Cluster, Receptor,Receptor.Cluster and MeanLR/another
measure" />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



<!-- dependencies from examples -->


  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CrossTalkeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/CrossTalkeR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/html/Output_Samples.html">Single Phenotype Report</a>
    </li>
    <li>
      <a href="../articles/html/Output_Samples1.html">Comparative Phenotype Report</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/CostaLab/CrossTalkeR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run all LR Downstream analysis</h1>
    <small class="dont-index">Source: <a href='https://github.com/CostaLab/CrossTalkeR/blob/master/R/generate_report.R'><code>R/generate_report.R</code></a></small>
    <div class="hidden name"><code>generate_report.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function loads the single conditions LR outputs and return the LR network based analysis.
It assumes that the table present the following columns Ligand, Ligand.Cluster, Receptor,Receptor.Cluster and MeanLR/another
measure</p>
    </div>

    <pre class="usage"><span class='fu'>generate_report</span><span class='op'>(</span>
  <span class='va'>lrpaths</span>,
  genes <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>out_path</span>,
  sep <span class='op'>=</span> <span class='st'>","</span>,
  threshold <span class='op'>=</span> <span class='fl'>0</span>,
  colors <span class='op'>=</span> <span class='cn'>NULL</span>,
  out_file <span class='op'>=</span> <span class='cn'>NULL</span>,
  report <span class='op'>=</span> <span class='cn'>TRUE</span>,
  output_fmt <span class='op'>=</span> <span class='st'>"html_document"</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>lrpaths</th>
      <td><p>Paths of single condition LR data</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p>list of genes to be considered in the sankey plots</p></td>
    </tr>
    <tr>
      <th>out_path</th>
      <td><p>output directory path</p></td>
    </tr>
    <tr>
      <th>sep</th>
      <td><p>character used on csv</p></td>
    </tr>
    <tr>
      <th>threshold</th>
      <td><p>percentage of edges to be pruned</p></td>
    </tr>
    <tr>
      <th>colors</th>
      <td><p>celltypes colorscheme</p></td>
    </tr>
    <tr>
      <th>out_file</th>
      <td><p>output file names</p></td>
    </tr>
    <tr>
      <th>report</th>
      <td><p>decide if a report is generated or not</p></td>
    </tr>
    <tr>
      <th>output_fmt</th>
      <td><p>rmarkdown render output format parameter</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Rmarkdown report all objects from each step</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>paths</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'CTR'</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>,
                              <span class='st'>"ctr_nils_bm_human.csv"</span>,
                              package <span class='op'>=</span> <span class='st'>"CrossTalkeR"</span><span class='op'>)</span>,
          <span class='st'>'EXP'</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>,
                              <span class='st'>"exp_nils_bm_human.csv"</span>,
                              package <span class='op'>=</span> <span class='st'>"CrossTalkeR"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>output</span> <span class='op'>=</span>  <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, package <span class='op'>=</span> <span class='st'>"CrossTalkeR"</span><span class='op'>)</span>
<span class='va'>genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>'TGFB1'</span><span class='op'>)</span>
<span class='va'>data</span> <span class='op'>&lt;-</span> <span class='fu'>generate_report</span><span class='op'>(</span>lrpaths <span class='op'>=</span> <span class='va'>paths</span>,
                       genes <span class='op'>=</span> <span class='va'>genes</span>,
                       out_path <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>output</span>,<span class='st'>'/'</span><span class='op'>)</span>,
                       threshold <span class='op'>=</span> <span class='fl'>0</span>,
                         out_file <span class='op'>=</span> <span class='st'>"report.html"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading Files</span></div><div class='output co'>#&gt;                                                                                  CTR 
#&gt; "/tmp/RtmpVbBFue/temp_libpath205728ee23d7/CrossTalkeR/extdata/ctr_nils_bm_human.csv" 
#&gt; [1] 5
#&gt;                                                                                  EXP 
#&gt; "/tmp/RtmpVbBFue/temp_libpath205728ee23d7/CrossTalkeR/extdata/exp_nils_bm_human.csv" 
#&gt; [1] 5</div><div class='output co'>#&gt; <span class='message'>Create a Differential Table</span></div><div class='output co'>#&gt; <span class='message'>Calculating CCI Ranking</span></div><div class='output co'>#&gt; <span class='message'>EXP_x_CTR</span></div><div class='output co'>#&gt; <span class='message'>Calculating GCI Ranking</span></div><div class='output co'>#&gt; <span class='message'>EXP_x_CTR</span></div><div class='output co'>#&gt; <span class='message'>Anntating the top Cell Genes</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>Reading KEGG annotation online:</span></div><div class='output co'>#&gt; <span class='message'>Reading KEGG annotation online:</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>'select()' returned 1:many mapping between keys and columns</span></div><div class='output co'>#&gt; <span class='message'>Defining templates</span></div><div class='output co'>#&gt; <span class='message'>Generating Report</span></div><div class='output co'>#&gt; <span class='message'>Preparing Single Phenotype Report</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: igraph</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'igraph'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:stats':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    decompose, spectrum</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    union</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: dplyr</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'dplyr'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:igraph':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    as_data_frame, groups, union</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:stats':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    filter, lag</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    intersect, setdiff, setequal, union</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: tibble</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'tibble'</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:igraph':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    as_data_frame</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: tidyr</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'tidyr'</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:igraph':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    crossing</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stringr</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: patchwork</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: factoextra</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: ggplot2</span></div><div class='output co'>#&gt; <span class='message'>Welcome! Want to learn more? See two factoextra-related books at https://goo.gl/ve3WBa</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: ComplexHeatmap</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: grid</span></div><div class='output co'>#&gt; <span class='message'>========================================</span>
#&gt; <span class='message'>ComplexHeatmap version 2.6.2</span>
#&gt; <span class='message'>Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span>
#&gt; <span class='message'>Github page: https://github.com/jokergoo/ComplexHeatmap</span>
#&gt; <span class='message'>Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>If you use it in published research, please cite:</span>
#&gt; <span class='message'>Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span>
#&gt; <span class='message'>  genomic data. Bioinformatics 2016.</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>This message can be suppressed by:</span>
#&gt; <span class='message'>  suppressPackageStartupMessages(library(ComplexHeatmap))</span>
#&gt; <span class='message'>========================================</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: DT</span></div><div class='output co'>#&gt; <span class='message'>Printing CCI CTR</span></div><div class='output co'>#&gt; <span class='message'>Printing CCI EXP</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: ggrepel</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 1515 rows containing missing values (geom_label_repel).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 1409 rows containing missing values (geom_label_repel).</span></div><div class='output co'>#&gt; <span class='message'>Preparing Comparative Phenotype Report</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: ggraph</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: graphlayouts</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: pals</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: extrafont</span></div><div class='output co'>#&gt; <span class='message'>Registering fonts with R</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: colorBlindness</span></div><div class='output co'>#&gt; <span class='message'>Printing CCI EXP_x_CTR</span></div><div class='output co'>#&gt; <span class='warning'>Warning: number of items to replace is not a multiple of replacement length</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 1636 rows containing missing values (geom_label_repel).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: ggrepel: 87 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></div><div class='output co'>#&gt; <span class='warning'>Warning: ggrepel: 87 unlabeled data points (too many overlaps). Consider increasing max.overlaps</span></div><div class='output co'>#&gt; <span class='message'>DONE!</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by James Shiniti Nagai, Nils Leimkühler, Michael T. Schaub, Rebekka Schneider, Ivan Costa Gesteira.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


